  INFO root [2019-09-14 09:49:40,955] - 
  INFO root [2019-09-14 09:49:40,955] - -----------------------------------------------------------------------------------------


  INFO root [2019-09-14 09:49:40,958] - Resmgr::initialize: KBE_ROOT=D:/kbengine/
  INFO root [2019-09-14 09:49:40,958] - Resmgr::initialize: KBE_RES_PATH=D:/kbengine/kbe/res/;D:/kbengine/BattleOfBalls_assets/;D:/kbengine/BattleOfBalls_assets/scripts/;D:/kbengine/BattleOfBalls_assets/res/
  INFO root [2019-09-14 09:49:40,958] - Resmgr::initialize: KBE_BIN_PATH=D:/kbengine/kbe/bin/server/
  INFO root [2019-09-14 09:49:40,978] - NetworkInterface::initialize(EXTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:20013 )
  INFO root [2019-09-14 09:49:40,978] - NetworkInterface::initialize(EXTERNAL-TCP): address 192.168.56.1:20013, SOMAXCONN=128.
  INFO root [2019-09-14 09:49:40,982] - NetworkInterface::initialize(INTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:64338 )
  INFO root [2019-09-14 09:49:40,982] - NetworkInterface::initialize(INTERNAL-TCP): address 192.168.56.1:64338, SOMAXCONN=128.
  INFO root [2019-09-14 09:49:40,982] - server-configs:
  INFO root [2019-09-14 09:49:40,983] - 	internalTcpAddr : 192.168.56.1:64338
  INFO root [2019-09-14 09:49:40,983] - 	externalTcpAddr : 192.168.56.1:20013
  INFO root [2019-09-14 09:49:40,983] - 	componentID : 9000
  WARN root [2019-09-14 09:49:41,184] - invalid UID(0) <= 0, please check UID for environment! automatically set to 2082.
  INFO root [2019-09-14 09:49:42,691] - ---- loginapp Version: 2.4.5. ScriptVersion: 0.1.0. Pythoncore: 3.7.3. Protocol: CE9FF53F4E85387EE4455C7191269324. Config: Debug X86. Built: 09:58:16 May  8 2019. AppID: 9000. UID: 2082. PID: 11468 ----
  INFO root [2019-09-14 09:49:42,691] - Server error_name: Intel(R) Core(TM) i5-6300HQ CPU @ 2.30GHz with 2097151kB RAM
  INFO root [2019-09-14 09:49:42,692] - ThreadPool::createThreadPool: creating  threadpool...
  INFO root [2019-09-14 09:49:42,692] - ThreadPool::createThreadPool: successfully(2), newThreadCount=1, normalMaxThreadCount=2, maxThreadCount=8
  INFO root [2019-09-14 09:49:42,883] - Script::install(): is successfully, Python=(3.7.3 (default, May  8 2019, 09:53:38) [MSC v.1916 32 bit (Intel)])!
  INFO root [2019-09-14 09:49:42,883] - Script::install(): KBExtra is successfully!
S_INFO root [2019-09-14 09:49:42,894] - onLoginAppReady: bootstrapGroupIndex=0, bootstrapGlobalIndex=0
  INFO root [2019-09-14 09:49:42,894] - TelnetServer server is running on port 31000
  INFO root [2019-09-14 09:49:43,294] - ---- loginapp is running ----
 DEBUG root [2019-09-14 09:49:43,323] - Components::process(): Request for the process of identity...
  INFO root [2019-09-14 09:49:50,454] - Components::findComponents: find logger(1)...
  INFO root [2019-09-14 09:49:50,550] - Components::findComponents: found logger, addr:192.168.56.1:64341
  INFO root [2019-09-14 09:49:50,550] - Components::addComponent: reset orderLog, uid=2082!
  INFO root [2019-09-14 09:49:50,551] - Components::addComponent[logger], uid=2082, componentID=2000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2019-09-14 09:49:50,552] - Components::findComponents: find dbmgr(2)...
  INFO root [2019-09-14 09:49:50,556] - Components::findComponents: found dbmgr, addr:192.168.56.1:64342
  INFO root [2019-09-14 09:49:50,556] - Components::addComponent: reset orderLog, uid=2082!
  INFO root [2019-09-14 09:49:50,556] - Components::addComponent[dbmgr], uid=2082, componentID=4000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2019-09-14 09:49:50,560] - Components::findComponents: found logger, addr:192.168.56.1:64341
  INFO root [2019-09-14 09:49:50,564] - Components::findComponents: found dbmgr, addr:192.168.56.1:64342
  WARN root [2019-09-14 09:49:51,565] - Forwarding logs to logger[192.168.56.1:64341/0/0/0]...
 ERROR root [2019-09-14 09:52:42,462] - Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=520045460868, lastReceivedTime=520042270469, inactivityExceptionPeriod=134887564)
  INFO root [2019-09-14 09:52:42,463] - The following logs sent to logger failed:
  WARN root [2019-09-14 09:52:42,464] - ==> [2019-09-14 09:52:42 462] TCPPacketReceiver::processPendingEvents(192.168.56.1:64341): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-14 09:52:42,464] - ==> [2019-09-14 09:52:42 462] Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=520045460868, lastReceivedTime=520042270469, inactivityExceptionPeriod=134887564)
  WARN root [2019-09-14 09:52:42,560] - TCPPacketReceiver::processPendingEvents(192.168.56.1:64342): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-14 09:52:42,561] - Components::removeComponentByChannel: dbmgr : 4000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=520045684653, lastReceivedTime=520042912023, inactivityExceptionPeriod=134887564)
  INFO root [2019-09-14 09:52:42,567] - Shutdowner::onShutdownBegin: shutting down(period=0, tickPeriod=1)
S_INFO root [2019-09-14 09:52:42,568] - onLoginAppShutDown()
  INFO root [2019-09-14 09:52:42,568] - Shutdowner::onShutdown: shutting down
  INFO root [2019-09-14 09:52:42,568] - Shutdowner::onShutdownEnd: shutting down
  INFO root [2019-09-14 09:52:42,578] - Script::uninstall(): is successfully!
 DEBUG root [2019-09-14 09:52:42,578] - ThreadPool::destroy(): starting size 2.
  WARN root [2019-09-14 09:52:42,879] - ThreadPool::destroy(): waiting for thread(2)[0x1f31880,0x1f318e8,], try=1
  INFO root [2019-09-14 09:52:44,162] - 
  INFO root [2019-09-14 09:52:44,163] - -----------------------------------------------------------------------------------------


  INFO root [2019-09-14 09:52:44,164] - Resmgr::initialize: KBE_ROOT=D:/kbengine/
  INFO root [2019-09-14 09:52:44,165] - Resmgr::initialize: KBE_RES_PATH=D:/kbengine/kbe/res/;D:/kbengine/BattleOfBalls_assets/;D:/kbengine/BattleOfBalls_assets/scripts/;D:/kbengine/BattleOfBalls_assets/res/
  INFO root [2019-09-14 09:52:44,165] - Resmgr::initialize: KBE_BIN_PATH=D:/kbengine/kbe/bin/server/
  INFO root [2019-09-14 09:52:44,230] - NetworkInterface::initialize(EXTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:20013 )
  INFO root [2019-09-14 09:52:44,230] - NetworkInterface::initialize(EXTERNAL-TCP): address 192.168.56.1:20013, SOMAXCONN=128.
  INFO root [2019-09-14 09:52:44,234] - NetworkInterface::initialize(INTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:64381 )
  INFO root [2019-09-14 09:52:44,234] - NetworkInterface::initialize(INTERNAL-TCP): address 192.168.56.1:64381, SOMAXCONN=128.
  INFO root [2019-09-14 09:52:44,235] - server-configs:
  INFO root [2019-09-14 09:52:44,235] - 	internalTcpAddr : 192.168.56.1:64381
  INFO root [2019-09-14 09:52:44,235] - 	externalTcpAddr : 192.168.56.1:20013
  INFO root [2019-09-14 09:52:44,235] - 	componentID : 9000
  WARN root [2019-09-14 09:52:44,481] - invalid UID(0) <= 0, please check UID for environment! automatically set to 2082.
  INFO root [2019-09-14 09:52:44,762] - Components::findLogger: found logger, addr:192.168.56.1:64376
  INFO root [2019-09-14 09:52:44,762] - Components::addComponent: reset orderLog, uid=2082!
  INFO root [2019-09-14 09:52:44,762] - Components::addComponent[logger], uid=2082, componentID=2000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2019-09-14 09:52:44,767] - ---- loginapp Version: 2.4.5. ScriptVersion: 0.1.0. Pythoncore: 3.7.3. Protocol: CE9FF53F4E85387EE4455C7191269324. Config: Debug X86. Built: 09:58:16 May  8 2019. AppID: 9000. UID: 2082. PID: 10304 ----
  INFO root [2019-09-14 09:52:44,767] - Server error_name: Intel(R) Core(TM) i5-6300HQ CPU @ 2.30GHz with 2097151kB RAM
  INFO root [2019-09-14 09:52:44,768] - ThreadPool::createThreadPool: creating  threadpool...
  INFO root [2019-09-14 09:52:44,768] - ThreadPool::createThreadPool: successfully(2), newThreadCount=1, normalMaxThreadCount=2, maxThreadCount=8
  INFO root [2019-09-14 09:52:44,966] - Script::install(): is successfully, Python=(3.7.3 (default, May  8 2019, 09:53:38) [MSC v.1916 32 bit (Intel)])!
  INFO root [2019-09-14 09:52:44,966] - Script::install(): KBExtra is successfully!
S_INFO root [2019-09-14 09:52:44,978] - onLoginAppReady: bootstrapGroupIndex=0, bootstrapGlobalIndex=0
  INFO root [2019-09-14 09:52:44,978] - TelnetServer server is running on port 31000
  INFO root [2019-09-14 09:52:45,352] - ---- loginapp is running ----
 DEBUG root [2019-09-14 09:52:45,352] - Components::process(): Request for the process of identity...
  WARN root [2019-09-14 09:52:51,483] - Forwarding logs to logger[192.168.56.1:64376/0/0/0]...
 ERROR root [2019-09-14 09:55:02,036] - Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=520359501904, lastReceivedTime=520319153335, inactivityExceptionPeriod=134887437)
  INFO root [2019-09-14 09:55:02,038] - The following logs sent to logger failed:
  WARN root [2019-09-14 09:55:02,038] - ==> [2019-09-14 09:55:02 035] TCPPacketReceiver::processPendingEvents(192.168.56.1:64376): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-14 09:55:02,038] - ==> [2019-09-14 09:55:02 036] Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=520359501904, lastReceivedTime=520319153335, inactivityExceptionPeriod=134887437)
  WARN root [2019-09-14 09:55:02,138] - TCPPacketReceiver::processPendingEvents(192.168.56.1:64377): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-14 09:55:02,142] - Components::removeComponentByChannel: dbmgr : 4000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=520359742290, lastReceivedTime=520319246569, inactivityExceptionPeriod=134887437)
  INFO root [2019-09-14 09:55:02,146] - Shutdowner::onShutdownBegin: shutting down(period=0, tickPeriod=1)
S_INFO root [2019-09-14 09:55:02,148] - onLoginAppShutDown()
  INFO root [2019-09-14 09:55:02,148] - Shutdowner::onShutdown: shutting down
  INFO root [2019-09-14 09:55:02,148] - Shutdowner::onShutdownEnd: shutting down
  INFO root [2019-09-14 09:55:02,160] - Script::uninstall(): is successfully!
 DEBUG root [2019-09-14 09:55:02,160] - ThreadPool::destroy(): starting size 2.
  WARN root [2019-09-14 09:55:02,462] - ThreadPool::destroy(): waiting for thread(2)[0x20328a0,0x2032908,], try=1
  INFO root [2019-09-14 10:00:43,082] - 
  INFO root [2019-09-14 10:00:43,098] - -----------------------------------------------------------------------------------------


  INFO root [2019-09-14 10:00:43,098] - Resmgr::initialize: KBE_ROOT=D:/kbengine/
  INFO root [2019-09-14 10:00:43,099] - Resmgr::initialize: KBE_RES_PATH=D:/kbengine/kbe/res/;D:/kbengine/BattleOfBalls_assets/;D:/kbengine/BattleOfBalls_assets/scripts/;D:/kbengine/BattleOfBalls_assets/res/
  INFO root [2019-09-14 10:00:43,099] - Resmgr::initialize: KBE_BIN_PATH=D:/kbengine/kbe/bin/server/
  INFO root [2019-09-14 10:00:43,152] - NetworkInterface::initialize(EXTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:20013 )
  INFO root [2019-09-14 10:00:43,152] - NetworkInterface::initialize(EXTERNAL-TCP): address 192.168.56.1:20013, SOMAXCONN=128.
  INFO root [2019-09-14 10:00:43,156] - NetworkInterface::initialize(INTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:64422 )
  INFO root [2019-09-14 10:00:43,159] - NetworkInterface::initialize(INTERNAL-TCP): address 192.168.56.1:64422, SOMAXCONN=128.
  INFO root [2019-09-14 10:00:43,160] - server-configs:
  INFO root [2019-09-14 10:00:43,161] - 	internalTcpAddr : 192.168.56.1:64422
  INFO root [2019-09-14 10:00:43,161] - 	externalTcpAddr : 192.168.56.1:20013
  INFO root [2019-09-14 10:00:43,161] - 	componentID : 9000
  WARN root [2019-09-14 10:00:43,387] - invalid UID(0) <= 0, please check UID for environment! automatically set to 2082.
  INFO root [2019-09-14 10:00:43,754] - Components::findLogger: found logger, addr:192.168.56.1:64418
  INFO root [2019-09-14 10:00:43,754] - Components::addComponent: reset orderLog, uid=2082!
  INFO root [2019-09-14 10:00:43,755] - Components::addComponent[logger], uid=2082, componentID=2000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2019-09-14 10:00:43,762] - ---- loginapp Version: 2.4.5. ScriptVersion: 0.1.0. Pythoncore: 3.7.3. Protocol: CE9FF53F4E85387EE4455C7191269324. Config: Debug X86. Built: 09:58:16 May  8 2019. AppID: 9000. UID: 2082. PID: 17872 ----
  INFO root [2019-09-14 10:00:43,763] - Server error_name: Intel(R) Core(TM) i5-6300HQ CPU @ 2.30GHz with 2097151kB RAM
  INFO root [2019-09-14 10:00:43,763] - ThreadPool::createThreadPool: creating  threadpool...
  INFO root [2019-09-14 10:00:43,763] - ThreadPool::createThreadPool: successfully(2), newThreadCount=1, normalMaxThreadCount=2, maxThreadCount=8
  INFO root [2019-09-14 10:00:43,978] - Script::install(): is successfully, Python=(3.7.3 (default, May  8 2019, 09:53:38) [MSC v.1916 32 bit (Intel)])!
  INFO root [2019-09-14 10:00:43,978] - Script::install(): KBExtra is successfully!
S_INFO root [2019-09-14 10:00:43,991] - onLoginAppReady: bootstrapGroupIndex=0, bootstrapGlobalIndex=0
  INFO root [2019-09-14 10:00:43,992] - TelnetServer server is running on port 31000
  INFO root [2019-09-14 10:00:44,430] - ---- loginapp is running ----
 DEBUG root [2019-09-14 10:00:44,431] - Components::process(): Request for the process of identity...
  WARN root [2019-09-14 10:00:50,549] - Forwarding logs to logger[192.168.56.1:64418/0/0/0]...
 ERROR root [2019-09-14 10:05:07,945] - Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=521722799142, lastReceivedTime=521666969432, inactivityExceptionPeriod=134887564)
  INFO root [2019-09-14 10:05:07,948] - The following logs sent to logger failed:
  WARN root [2019-09-14 10:05:07,948] - ==> [2019-09-14 10:05:07 944] TCPPacketReceiver::processPendingEvents(192.168.56.1:64418): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-14 10:05:07,948] - ==> [2019-09-14 10:05:07 946] Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=521722799142, lastReceivedTime=521666969432, inactivityExceptionPeriod=134887564)
  WARN root [2019-09-14 10:05:08,033] - TCPPacketReceiver::processPendingEvents(192.168.56.1:64417): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-14 10:05:08,035] - Components::removeComponentByChannel: dbmgr : 4000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=521723002027, lastReceivedTime=521666917528, inactivityExceptionPeriod=134887564)
  INFO root [2019-09-14 10:05:08,039] - Shutdowner::onShutdownBegin: shutting down(period=0, tickPeriod=1)
S_INFO root [2019-09-14 10:05:08,041] - onLoginAppShutDown()
  INFO root [2019-09-14 10:05:08,041] - Shutdowner::onShutdown: shutting down
  INFO root [2019-09-14 10:05:08,041] - Shutdowner::onShutdownEnd: shutting down
  INFO root [2019-09-14 10:05:08,049] - Script::uninstall(): is successfully!
 DEBUG root [2019-09-14 10:05:08,050] - ThreadPool::destroy(): starting size 2.
  WARN root [2019-09-14 10:05:08,351] - ThreadPool::destroy(): waiting for thread(2)[0x2173ce0,0x2174158,], try=1
  INFO root [2019-09-14 10:07:03,058] - 
  INFO root [2019-09-14 10:07:03,060] - -----------------------------------------------------------------------------------------


  INFO root [2019-09-14 10:07:03,061] - Resmgr::initialize: KBE_ROOT=D:/kbengine/
  INFO root [2019-09-14 10:07:03,061] - Resmgr::initialize: KBE_RES_PATH=D:/kbengine/kbe/res/;D:/kbengine/BattleOfBalls_assets/;D:/kbengine/BattleOfBalls_assets/scripts/;D:/kbengine/BattleOfBalls_assets/res/
  INFO root [2019-09-14 10:07:03,061] - Resmgr::initialize: KBE_BIN_PATH=D:/kbengine/kbe/bin/server/
  INFO root [2019-09-14 10:07:03,078] - NetworkInterface::initialize(EXTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:20013 )
  INFO root [2019-09-14 10:07:03,078] - NetworkInterface::initialize(EXTERNAL-TCP): address 192.168.56.1:20013, SOMAXCONN=128.
  INFO root [2019-09-14 10:07:03,083] - NetworkInterface::initialize(INTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:64463 )
  INFO root [2019-09-14 10:07:03,083] - NetworkInterface::initialize(INTERNAL-TCP): address 192.168.56.1:64463, SOMAXCONN=128.
  INFO root [2019-09-14 10:07:03,083] - server-configs:
  INFO root [2019-09-14 10:07:03,083] - 	internalTcpAddr : 192.168.56.1:64463
  INFO root [2019-09-14 10:07:03,083] - 	externalTcpAddr : 192.168.56.1:20013
  INFO root [2019-09-14 10:07:03,083] - 	componentID : 9000
  WARN root [2019-09-14 10:07:03,289] - invalid UID(0) <= 0, please check UID for environment! automatically set to 2082.
  INFO root [2019-09-14 10:07:03,464] - Components::findLogger: found logger, addr:192.168.56.1:64460
  INFO root [2019-09-14 10:07:03,464] - Components::addComponent: reset orderLog, uid=2082!
  INFO root [2019-09-14 10:07:03,464] - Components::addComponent[logger], uid=2082, componentID=2000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2019-09-14 10:07:03,469] - ---- loginapp Version: 2.4.5. ScriptVersion: 0.1.0. Pythoncore: 3.7.3. Protocol: CE9FF53F4E85387EE4455C7191269324. Config: Debug X86. Built: 09:58:16 May  8 2019. AppID: 9000. UID: 2082. PID: 10172 ----
  INFO root [2019-09-14 10:07:03,470] - Server error_name: Intel(R) Core(TM) i5-6300HQ CPU @ 2.30GHz with 2097151kB RAM
  INFO root [2019-09-14 10:07:03,470] - ThreadPool::createThreadPool: creating  threadpool...
  INFO root [2019-09-14 10:07:03,470] - ThreadPool::createThreadPool: successfully(2), newThreadCount=1, normalMaxThreadCount=2, maxThreadCount=8
  INFO root [2019-09-14 10:07:03,661] - Script::install(): is successfully, Python=(3.7.3 (default, May  8 2019, 09:53:38) [MSC v.1916 32 bit (Intel)])!
  INFO root [2019-09-14 10:07:03,661] - Script::install(): KBExtra is successfully!
S_INFO root [2019-09-14 10:07:03,671] - onLoginAppReady: bootstrapGroupIndex=0, bootstrapGlobalIndex=0
  INFO root [2019-09-14 10:07:03,672] - TelnetServer server is running on port 31000
  INFO root [2019-09-14 10:07:04,044] - ---- loginapp is running ----
 DEBUG root [2019-09-14 10:07:04,045] - Components::process(): Request for the process of identity...
  WARN root [2019-09-14 10:07:10,139] - Forwarding logs to logger[192.168.56.1:64460/0/0/0]...
 ERROR root [2019-09-14 10:14:27,404] - Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=522981580748, lastReceivedTime=522926691440, inactivityExceptionPeriod=134887437)
  INFO root [2019-09-14 10:14:27,406] - The following logs sent to logger failed:
  WARN root [2019-09-14 10:14:27,406] - ==> [2019-09-14 10:14:27 402] TCPPacketReceiver::processPendingEvents(192.168.56.1:64460): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-14 10:14:27,407] - ==> [2019-09-14 10:14:27 404] Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=522981580748, lastReceivedTime=522926691440, inactivityExceptionPeriod=134887437)
  WARN root [2019-09-14 10:14:27,496] - TCPPacketReceiver::processPendingEvents(192.168.56.1:64462): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-14 10:14:27,498] - Components::removeComponentByChannel: dbmgr : 4000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=522981792712, lastReceivedTime=522926817537, inactivityExceptionPeriod=134887437)
  INFO root [2019-09-14 10:14:27,506] - Shutdowner::onShutdownBegin: shutting down(period=0, tickPeriod=1)
S_INFO root [2019-09-14 10:14:27,509] - onLoginAppShutDown()
  INFO root [2019-09-14 10:14:27,509] - Shutdowner::onShutdown: shutting down
  INFO root [2019-09-14 10:14:27,509] - Shutdowner::onShutdownEnd: shutting down
  INFO root [2019-09-14 10:14:27,518] - Script::uninstall(): is successfully!
 DEBUG root [2019-09-14 10:14:27,518] - ThreadPool::destroy(): starting size 2.
  WARN root [2019-09-14 10:14:27,819] - ThreadPool::destroy(): waiting for thread(2)[0x2185390,0x2185530,], try=1
  INFO root [2019-09-14 10:14:29,095] - 
  INFO root [2019-09-14 10:14:29,095] - -----------------------------------------------------------------------------------------


  INFO root [2019-09-14 10:14:29,096] - Resmgr::initialize: KBE_ROOT=D:/kbengine/
  INFO root [2019-09-14 10:14:29,097] - Resmgr::initialize: KBE_RES_PATH=D:/kbengine/kbe/res/;D:/kbengine/BattleOfBalls_assets/;D:/kbengine/BattleOfBalls_assets/scripts/;D:/kbengine/BattleOfBalls_assets/res/
  INFO root [2019-09-14 10:14:29,097] - Resmgr::initialize: KBE_BIN_PATH=D:/kbengine/kbe/bin/server/
  INFO root [2019-09-14 10:14:29,114] - NetworkInterface::initialize(EXTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:20013 )
  INFO root [2019-09-14 10:14:29,114] - NetworkInterface::initialize(EXTERNAL-TCP): address 192.168.56.1:20013, SOMAXCONN=128.
  INFO root [2019-09-14 10:14:29,121] - NetworkInterface::initialize(INTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:64501 )
  INFO root [2019-09-14 10:14:29,121] - NetworkInterface::initialize(INTERNAL-TCP): address 192.168.56.1:64501, SOMAXCONN=128.
  INFO root [2019-09-14 10:14:29,121] - server-configs:
  INFO root [2019-09-14 10:14:29,122] - 	internalTcpAddr : 192.168.56.1:64501
  INFO root [2019-09-14 10:14:29,122] - 	externalTcpAddr : 192.168.56.1:20013
  INFO root [2019-09-14 10:14:29,122] - 	componentID : 9000
  WARN root [2019-09-14 10:14:29,347] - invalid UID(0) <= 0, please check UID for environment! automatically set to 2082.
  INFO root [2019-09-14 10:14:30,853] - ---- loginapp Version: 2.4.5. ScriptVersion: 0.1.0. Pythoncore: 3.7.3. Protocol: CE9FF53F4E85387EE4455C7191269324. Config: Debug X86. Built: 09:58:16 May  8 2019. AppID: 9000. UID: 2082. PID: 12504 ----
  INFO root [2019-09-14 10:14:30,879] - Server error_name: Intel(R) Core(TM) i5-6300HQ CPU @ 2.30GHz with 2097151kB RAM
  INFO root [2019-09-14 10:14:30,880] - ThreadPool::createThreadPool: creating  threadpool...
  INFO root [2019-09-14 10:14:30,880] - ThreadPool::createThreadPool: successfully(2), newThreadCount=1, normalMaxThreadCount=2, maxThreadCount=8
  INFO root [2019-09-14 10:14:31,125] - Script::install(): is successfully, Python=(3.7.3 (default, May  8 2019, 09:53:38) [MSC v.1916 32 bit (Intel)])!
  INFO root [2019-09-14 10:14:31,126] - Script::install(): KBExtra is successfully!
S_INFO root [2019-09-14 10:14:31,137] - onLoginAppReady: bootstrapGroupIndex=0, bootstrapGlobalIndex=0
  INFO root [2019-09-14 10:14:31,137] - TelnetServer server is running on port 31000
  INFO root [2019-09-14 10:14:31,560] - ---- loginapp is running ----
 DEBUG root [2019-09-14 10:14:31,586] - Components::process(): Request for the process of identity...
  INFO root [2019-09-14 10:14:38,717] - Components::findComponents: find logger(1)...
  INFO root [2019-09-14 10:14:38,803] - Components::findComponents: found logger, addr:192.168.56.1:64497
  INFO root [2019-09-14 10:14:38,803] - Components::addComponent: reset orderLog, uid=2082!
  INFO root [2019-09-14 10:14:38,803] - Components::addComponent[logger], uid=2082, componentID=2000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2019-09-14 10:14:38,805] - Components::findComponents: find dbmgr(2)...
  INFO root [2019-09-14 10:14:38,812] - Components::findComponents: found logger, addr:192.168.56.1:64497
  INFO root [2019-09-14 10:14:38,815] - Components::findComponents: found dbmgr, addr:192.168.56.1:64500
  INFO root [2019-09-14 10:14:38,815] - Components::addComponent: reset orderLog, uid=2082!
  INFO root [2019-09-14 10:14:38,815] - Components::addComponent[dbmgr], uid=2082, componentID=4000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2019-09-14 10:14:38,916] - Components::findComponents: found dbmgr, addr:192.168.56.1:64500
  WARN root [2019-09-14 10:14:39,917] - Forwarding logs to logger[192.168.56.1:64497/0/0/0]...
 ERROR root [2019-09-14 10:18:54,507] - Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=523582563353, lastReceivedTime=523525344257, inactivityExceptionPeriod=134887564)
  INFO root [2019-09-14 10:18:54,511] - The following logs sent to logger failed:
  WARN root [2019-09-14 10:18:54,511] - ==> [2019-09-14 10:18:54 504] TCPPacketReceiver::processPendingEvents(192.168.56.1:64497): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-14 10:18:54,512] - ==> [2019-09-14 10:18:54 507] Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=523582563353, lastReceivedTime=523525344257, inactivityExceptionPeriod=134887564)
  WARN root [2019-09-14 10:18:54,593] - TCPPacketReceiver::processPendingEvents(192.168.56.1:64500): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-14 10:18:54,594] - Components::removeComponentByChannel: dbmgr : 4000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=523582760540, lastReceivedTime=523525436880, inactivityExceptionPeriod=134887564)
  INFO root [2019-09-14 10:18:54,595] - Shutdowner::onShutdownBegin: shutting down(period=0, tickPeriod=1)
S_INFO root [2019-09-14 10:18:54,595] - onLoginAppShutDown()
  INFO root [2019-09-14 10:18:54,596] - Shutdowner::onShutdown: shutting down
  INFO root [2019-09-14 10:18:54,596] - Shutdowner::onShutdownEnd: shutting down
  INFO root [2019-09-14 10:18:54,604] - Script::uninstall(): is successfully!
 DEBUG root [2019-09-14 10:18:54,605] - ThreadPool::destroy(): starting size 2.
  WARN root [2019-09-14 10:18:54,906] - ThreadPool::destroy(): waiting for thread(2)[0x2129330,0x2128ff0,], try=1
  INFO root [2019-09-14 15:04:38,559] - 
  INFO root [2019-09-14 15:04:38,571] - -----------------------------------------------------------------------------------------


  INFO root [2019-09-14 15:04:38,572] - Resmgr::initialize: KBE_ROOT=D:/kbengine/
  INFO root [2019-09-14 15:04:38,572] - Resmgr::initialize: KBE_RES_PATH=D:/kbengine/kbe/res/;D:/kbengine/BattleOfBalls_assets/;D:/kbengine/BattleOfBalls_assets/scripts/;D:/kbengine/BattleOfBalls_assets/res/
  INFO root [2019-09-14 15:04:38,573] - Resmgr::initialize: KBE_BIN_PATH=D:/kbengine/kbe/bin/server/
  INFO root [2019-09-14 15:04:38,642] - NetworkInterface::initialize(EXTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:20013 )
  INFO root [2019-09-14 15:04:38,643] - NetworkInterface::initialize(EXTERNAL-TCP): address 192.168.56.1:20013, SOMAXCONN=128.
  INFO root [2019-09-14 15:04:38,658] - NetworkInterface::initialize(INTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:51065 )
  INFO root [2019-09-14 15:04:38,658] - NetworkInterface::initialize(INTERNAL-TCP): address 192.168.56.1:51065, SOMAXCONN=128.
  INFO root [2019-09-14 15:04:38,659] - server-configs:
  INFO root [2019-09-14 15:04:38,659] - 	internalTcpAddr : 192.168.56.1:51065
  INFO root [2019-09-14 15:04:38,659] - 	externalTcpAddr : 192.168.56.1:20013
  INFO root [2019-09-14 15:04:38,659] - 	componentID : 9000
  WARN root [2019-09-14 15:04:38,887] - invalid UID(0) <= 0, please check UID for environment! automatically set to 2082.
  INFO root [2019-09-14 15:04:38,907] - Components::findLogger: found logger, addr:192.168.56.1:51060
  INFO root [2019-09-14 15:04:38,908] - Components::addComponent: reset orderLog, uid=2082!
  INFO root [2019-09-14 15:04:38,908] - Components::addComponent[logger], uid=2082, componentID=2000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2019-09-14 15:04:44,920] - ---- loginapp Version: 2.4.5. ScriptVersion: 0.1.0. Pythoncore: 3.7.3. Protocol: CE9FF53F4E85387EE4455C7191269324. Config: Debug X86. Built: 09:58:16 May  8 2019. AppID: 9000. UID: 2082. PID: 13444 ----
  INFO root [2019-09-14 15:04:44,921] - Server error_name: Intel(R) Core(TM) i5-6300HQ CPU @ 2.30GHz with 2097151kB RAM
  INFO root [2019-09-14 15:04:44,922] - ThreadPool::createThreadPool: creating  threadpool...
  INFO root [2019-09-14 15:04:44,922] - ThreadPool::createThreadPool: successfully(2), newThreadCount=1, normalMaxThreadCount=2, maxThreadCount=8
  INFO root [2019-09-14 15:04:45,125] - Script::install(): is successfully, Python=(3.7.3 (default, May  8 2019, 09:53:38) [MSC v.1916 32 bit (Intel)])!
  INFO root [2019-09-14 15:04:45,125] - Script::install(): KBExtra is successfully!
S_INFO root [2019-09-14 15:04:45,137] - onLoginAppReady: bootstrapGroupIndex=0, bootstrapGlobalIndex=0
  INFO root [2019-09-14 15:04:45,138] - TelnetServer server is running on port 31000
  INFO root [2019-09-14 15:04:45,538] - ---- loginapp is running ----
 DEBUG root [2019-09-14 15:04:45,539] - Components::process(): Request for the process of identity...
  INFO root [2019-09-14 15:04:52,658] - Components::findComponents: find logger(1)...
  INFO root [2019-09-14 15:04:52,748] - Components::findComponents: found logger, addr:192.168.56.1:51060
  INFO root [2019-09-14 15:04:52,750] - Components::findComponents: find dbmgr(2)...
  WARN root [2019-09-14 15:04:53,853] - Forwarding logs to logger[192.168.56.1:51060/0/0/0]...
 ERROR root [2019-09-14 15:08:24,957] - Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=30040469211, lastReceivedTime=30003125017, inactivityExceptionPeriod=134887628)
  INFO root [2019-09-14 15:08:24,966] - The following logs sent to logger failed:
  WARN root [2019-09-14 15:08:24,967] - ==> [2019-09-14 15:08:24 953] TCPPacketReceiver::processPendingEvents(192.168.56.1:51060): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-14 15:08:24,967] - ==> [2019-09-14 15:08:24 958] Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=30040469211, lastReceivedTime=30003125017, inactivityExceptionPeriod=134887628)
  WARN root [2019-09-14 15:08:25,082] - TCPPacketReceiver::processPendingEvents(192.168.56.1:51061): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-14 15:08:25,087] - Components::removeComponentByChannel: dbmgr : 4000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=30040761285, lastReceivedTime=30003211054, inactivityExceptionPeriod=134887628)
  INFO root [2019-09-14 15:08:25,114] - Shutdowner::onShutdownBegin: shutting down(period=0, tickPeriod=1)
S_INFO root [2019-09-14 15:08:25,122] - onLoginAppShutDown()
  INFO root [2019-09-14 15:08:25,122] - Shutdowner::onShutdown: shutting down
  INFO root [2019-09-14 15:08:25,122] - Shutdowner::onShutdownEnd: shutting down
  INFO root [2019-09-14 15:08:25,143] - Script::uninstall(): is successfully!
 DEBUG root [2019-09-14 15:08:25,144] - ThreadPool::destroy(): starting size 2.
  WARN root [2019-09-14 15:08:25,445] - ThreadPool::destroy(): waiting for thread(2)[0x1c72658,0x1c726c0,], try=1
  INFO root [2019-09-14 15:11:38,913] - 
  INFO root [2019-09-14 15:11:38,922] - -----------------------------------------------------------------------------------------


  INFO root [2019-09-14 15:11:38,924] - Resmgr::initialize: KBE_ROOT=D:/kbengine/
  INFO root [2019-09-14 15:11:38,925] - Resmgr::initialize: KBE_RES_PATH=D:/kbengine/kbe/res/;D:/kbengine/BattleOfBalls_assets/;D:/kbengine/BattleOfBalls_assets/scripts/;D:/kbengine/BattleOfBalls_assets/res/
  INFO root [2019-09-14 15:11:38,926] - Resmgr::initialize: KBE_BIN_PATH=D:/kbengine/kbe/bin/server/
  INFO root [2019-09-14 15:11:38,968] - NetworkInterface::initialize(EXTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:20013 )
  INFO root [2019-09-14 15:11:38,968] - NetworkInterface::initialize(EXTERNAL-TCP): address 192.168.56.1:20013, SOMAXCONN=128.
  INFO root [2019-09-14 15:11:38,972] - NetworkInterface::initialize(INTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:51195 )
  INFO root [2019-09-14 15:11:38,973] - NetworkInterface::initialize(INTERNAL-TCP): address 192.168.56.1:51195, SOMAXCONN=128.
  INFO root [2019-09-14 15:11:38,973] - server-configs:
  INFO root [2019-09-14 15:11:38,973] - 	internalTcpAddr : 192.168.56.1:51195
  INFO root [2019-09-14 15:11:38,973] - 	externalTcpAddr : 192.168.56.1:20013
  INFO root [2019-09-14 15:11:38,973] - 	componentID : 9000
  WARN root [2019-09-14 15:11:39,223] - invalid UID(0) <= 0, please check UID for environment! automatically set to 2082.
  INFO root [2019-09-14 15:11:39,474] - Components::findLogger: found logger, addr:192.168.56.1:51190
  INFO root [2019-09-14 15:11:39,475] - Components::addComponent: reset orderLog, uid=2082!
  INFO root [2019-09-14 15:11:39,475] - Components::addComponent[logger], uid=2082, componentID=2000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2019-09-14 15:11:39,480] - ---- loginapp Version: 2.4.5. ScriptVersion: 0.1.0. Pythoncore: 3.7.3. Protocol: CE9FF53F4E85387EE4455C7191269324. Config: Debug X86. Built: 09:58:16 May  8 2019. AppID: 9000. UID: 2082. PID: 14024 ----
  INFO root [2019-09-14 15:11:39,480] - Server error_name: Intel(R) Core(TM) i5-6300HQ CPU @ 2.30GHz with 2097151kB RAM
  INFO root [2019-09-14 15:11:39,480] - ThreadPool::createThreadPool: creating  threadpool...
  INFO root [2019-09-14 15:11:39,481] - ThreadPool::createThreadPool: successfully(2), newThreadCount=1, normalMaxThreadCount=2, maxThreadCount=8
  INFO root [2019-09-14 15:11:39,701] - Script::install(): is successfully, Python=(3.7.3 (default, May  8 2019, 09:53:38) [MSC v.1916 32 bit (Intel)])!
  INFO root [2019-09-14 15:11:39,701] - Script::install(): KBExtra is successfully!
S_INFO root [2019-09-14 15:11:39,714] - onLoginAppReady: bootstrapGroupIndex=0, bootstrapGlobalIndex=0
  INFO root [2019-09-14 15:11:39,714] - TelnetServer server is running on port 31000
  INFO root [2019-09-14 15:11:40,150] - ---- loginapp is running ----
 DEBUG root [2019-09-14 15:11:40,151] - Components::process(): Request for the process of identity...
  WARN root [2019-09-14 15:11:46,290] - Forwarding logs to logger[192.168.56.1:51190/0/0/0]...
 ERROR root [2019-09-14 15:13:58,944] - Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=30791941083, lastReceivedTime=30746880136, inactivityExceptionPeriod=134887197)
  INFO root [2019-09-14 15:13:58,946] - The following logs sent to logger failed:
  WARN root [2019-09-14 15:13:58,946] - ==> [2019-09-14 15:13:58 937] TCPPacketReceiver::processPendingEvents(192.168.56.1:51190): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-14 15:13:58,946] - ==> [2019-09-14 15:13:58 945] Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=30791941083, lastReceivedTime=30746880136, inactivityExceptionPeriod=134887197)
  WARN root [2019-09-14 15:13:59,047] - TCPPacketReceiver::processPendingEvents(192.168.56.1:51192): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-14 15:13:59,049] - Components::removeComponentByChannel: dbmgr : 4000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=30792177415, lastReceivedTime=30746985842, inactivityExceptionPeriod=134887197)
  INFO root [2019-09-14 15:13:59,050] - Shutdowner::onShutdownBegin: shutting down(period=0, tickPeriod=1)
S_INFO root [2019-09-14 15:13:59,051] - onLoginAppShutDown()
  INFO root [2019-09-14 15:13:59,051] - Shutdowner::onShutdown: shutting down
  INFO root [2019-09-14 15:13:59,051] - Shutdowner::onShutdownEnd: shutting down
  INFO root [2019-09-14 15:13:59,062] - Script::uninstall(): is successfully!
 DEBUG root [2019-09-14 15:13:59,062] - ThreadPool::destroy(): starting size 2.
  WARN root [2019-09-14 15:13:59,363] - ThreadPool::destroy(): waiting for thread(2)[0x1e56380,0x1e568c8,], try=1
  INFO root [2019-09-14 15:16:28,263] - 
  INFO root [2019-09-14 15:16:28,268] - -----------------------------------------------------------------------------------------


  INFO root [2019-09-14 15:16:28,269] - Resmgr::initialize: KBE_ROOT=D:/kbengine/
  INFO root [2019-09-14 15:16:28,270] - Resmgr::initialize: KBE_RES_PATH=D:/kbengine/kbe/res/;D:/kbengine/BattleOfBalls_assets/;D:/kbengine/BattleOfBalls_assets/scripts/;D:/kbengine/BattleOfBalls_assets/res/
  INFO root [2019-09-14 15:16:28,270] - Resmgr::initialize: KBE_BIN_PATH=D:/kbengine/kbe/bin/server/
  INFO root [2019-09-14 15:16:28,310] - NetworkInterface::initialize(EXTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:20013 )
  INFO root [2019-09-14 15:16:28,311] - NetworkInterface::initialize(EXTERNAL-TCP): address 192.168.56.1:20013, SOMAXCONN=128.
  INFO root [2019-09-14 15:16:28,315] - NetworkInterface::initialize(INTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:51291 )
  INFO root [2019-09-14 15:16:28,315] - NetworkInterface::initialize(INTERNAL-TCP): address 192.168.56.1:51291, SOMAXCONN=128.
  INFO root [2019-09-14 15:16:28,315] - server-configs:
  INFO root [2019-09-14 15:16:28,315] - 	internalTcpAddr : 192.168.56.1:51291
  INFO root [2019-09-14 15:16:28,315] - 	externalTcpAddr : 192.168.56.1:20013
  INFO root [2019-09-14 15:16:28,315] - 	componentID : 9000
  WARN root [2019-09-14 15:16:28,535] - invalid UID(0) <= 0, please check UID for environment! automatically set to 2082.
  INFO root [2019-09-14 15:16:28,936] - Components::findLogger: found logger, addr:192.168.56.1:51288
  INFO root [2019-09-14 15:16:28,936] - Components::addComponent: reset orderLog, uid=2082!
  INFO root [2019-09-14 15:16:28,936] - Components::addComponent[logger], uid=2082, componentID=2000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2019-09-14 15:16:28,942] - ---- loginapp Version: 2.4.5. ScriptVersion: 0.1.0. Pythoncore: 3.7.3. Protocol: CE9FF53F4E85387EE4455C7191269324. Config: Debug X86. Built: 09:58:16 May  8 2019. AppID: 9000. UID: 2082. PID: 1164 ----
  INFO root [2019-09-14 15:16:28,943] - Server error_name: Intel(R) Core(TM) i5-6300HQ CPU @ 2.30GHz with 2097151kB RAM
  INFO root [2019-09-14 15:16:28,948] - ThreadPool::createThreadPool: creating  threadpool...
  INFO root [2019-09-14 15:16:28,950] - ThreadPool::createThreadPool: successfully(2), newThreadCount=1, normalMaxThreadCount=2, maxThreadCount=8
  INFO root [2019-09-14 15:16:29,187] - Script::install(): is successfully, Python=(3.7.3 (default, May  8 2019, 09:53:38) [MSC v.1916 32 bit (Intel)])!
  INFO root [2019-09-14 15:16:29,188] - Script::install(): KBExtra is successfully!
S_INFO root [2019-09-14 15:16:29,205] - onLoginAppReady: bootstrapGroupIndex=0, bootstrapGlobalIndex=0
  INFO root [2019-09-14 15:16:29,205] - TelnetServer server is running on port 31000
  INFO root [2019-09-14 15:16:29,619] - ---- loginapp is running ----
 DEBUG root [2019-09-14 15:16:29,619] - Components::process(): Request for the process of identity...
  WARN root [2019-09-14 15:16:35,769] - Forwarding logs to logger[192.168.56.1:51288/0/0/0]...
 ERROR root [2019-09-14 15:19:55,213] - Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=31593544179, lastReceivedTime=31532842886, inactivityExceptionPeriod=134887564)
  INFO root [2019-09-14 15:19:55,216] - The following logs sent to logger failed:
  WARN root [2019-09-14 15:19:55,217] - ==> [2019-09-14 15:19:55 209] TCPPacketReceiver::processPendingEvents(192.168.56.1:51288): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-14 15:19:55,217] - ==> [2019-09-14 15:19:55 213] Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=31593544179, lastReceivedTime=31532842886, inactivityExceptionPeriod=134887564)
  WARN root [2019-09-14 15:19:55,341] - TCPPacketReceiver::processPendingEvents(192.168.56.1:51290): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-14 15:19:55,346] - Components::removeComponentByChannel: dbmgr : 4000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=31593844585, lastReceivedTime=31532925009, inactivityExceptionPeriod=134887564)
  INFO root [2019-09-14 15:19:55,352] - Shutdowner::onShutdownBegin: shutting down(period=0, tickPeriod=1)
S_INFO root [2019-09-14 15:19:55,355] - onLoginAppShutDown()
  INFO root [2019-09-14 15:19:55,355] - Shutdowner::onShutdown: shutting down
  INFO root [2019-09-14 15:19:55,355] - Shutdowner::onShutdownEnd: shutting down
  INFO root [2019-09-14 15:19:55,367] - Script::uninstall(): is successfully!
 DEBUG root [2019-09-14 15:19:55,367] - ThreadPool::destroy(): starting size 2.
  WARN root [2019-09-14 15:19:55,668] - ThreadPool::destroy(): waiting for thread(2)[0xbe22d8,0xbe2000,], try=1
  INFO root [2019-09-14 15:20:11,979] - 
  INFO root [2019-09-14 15:20:11,987] - -----------------------------------------------------------------------------------------


  INFO root [2019-09-14 15:20:11,988] - Resmgr::initialize: KBE_ROOT=D:/kbengine/
  INFO root [2019-09-14 15:20:11,988] - Resmgr::initialize: KBE_RES_PATH=D:/kbengine/kbe/res/;D:/kbengine/BattleOfBalls_assets/;D:/kbengine/BattleOfBalls_assets/scripts/;D:/kbengine/BattleOfBalls_assets/res/
  INFO root [2019-09-14 15:20:11,989] - Resmgr::initialize: KBE_BIN_PATH=D:/kbengine/kbe/bin/server/
  INFO root [2019-09-14 15:20:12,021] - NetworkInterface::initialize(EXTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:20013 )
  INFO root [2019-09-14 15:20:12,021] - NetworkInterface::initialize(EXTERNAL-TCP): address 192.168.56.1:20013, SOMAXCONN=128.
  INFO root [2019-09-14 15:20:12,031] - NetworkInterface::initialize(INTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:51334 )
  INFO root [2019-09-14 15:20:12,032] - NetworkInterface::initialize(INTERNAL-TCP): address 192.168.56.1:51334, SOMAXCONN=128.
  INFO root [2019-09-14 15:20:12,032] - server-configs:
  INFO root [2019-09-14 15:20:12,032] - 	internalTcpAddr : 192.168.56.1:51334
  INFO root [2019-09-14 15:20:12,032] - 	externalTcpAddr : 192.168.56.1:20013
  INFO root [2019-09-14 15:20:12,032] - 	componentID : 9000
  WARN root [2019-09-14 15:20:12,247] - invalid UID(0) <= 0, please check UID for environment! automatically set to 2082.
  INFO root [2019-09-14 15:20:12,524] - Components::findLogger: found logger, addr:192.168.56.1:51327
  INFO root [2019-09-14 15:20:12,525] - Components::addComponent: reset orderLog, uid=2082!
  INFO root [2019-09-14 15:20:12,525] - Components::addComponent[logger], uid=2082, componentID=2000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2019-09-14 15:20:12,530] - ---- loginapp Version: 2.4.5. ScriptVersion: 0.1.0. Pythoncore: 3.7.3. Protocol: CE9FF53F4E85387EE4455C7191269324. Config: Debug X86. Built: 09:58:16 May  8 2019. AppID: 9000. UID: 2082. PID: 9748 ----
  INFO root [2019-09-14 15:20:12,530] - Server error_name: Intel(R) Core(TM) i5-6300HQ CPU @ 2.30GHz with 2097151kB RAM
  INFO root [2019-09-14 15:20:12,530] - ThreadPool::createThreadPool: creating  threadpool...
  INFO root [2019-09-14 15:20:12,531] - ThreadPool::createThreadPool: successfully(2), newThreadCount=1, normalMaxThreadCount=2, maxThreadCount=8
  INFO root [2019-09-14 15:20:12,768] - Script::install(): is successfully, Python=(3.7.3 (default, May  8 2019, 09:53:38) [MSC v.1916 32 bit (Intel)])!
  INFO root [2019-09-14 15:20:12,769] - Script::install(): KBExtra is successfully!
S_INFO root [2019-09-14 15:20:12,784] - onLoginAppReady: bootstrapGroupIndex=0, bootstrapGlobalIndex=0
  INFO root [2019-09-14 15:20:12,785] - TelnetServer server is running on port 31000
  INFO root [2019-09-14 15:20:13,190] - ---- loginapp is running ----
 DEBUG root [2019-09-14 15:20:13,191] - Components::process(): Request for the process of identity...
  WARN root [2019-09-14 15:20:19,320] - Forwarding logs to logger[192.168.56.1:51327/0/0/0]...
 ERROR root [2019-09-14 15:22:46,984] - Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=31980030714, lastReceivedTime=31968624733, inactivityExceptionPeriod=134887628)
  INFO root [2019-09-14 15:22:46,986] - The following logs sent to logger failed:
  WARN root [2019-09-14 15:22:46,986] - ==> [2019-09-14 15:22:46 984] TCPPacketReceiver::processPendingEvents(192.168.56.1:51327): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-14 15:22:46,987] - ==> [2019-09-14 15:22:46 985] Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=31980030714, lastReceivedTime=31968624733, inactivityExceptionPeriod=134887628)
  WARN root [2019-09-14 15:22:47,082] - TCPPacketReceiver::processPendingEvents(192.168.56.1:51332): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-14 15:22:47,087] - Components::removeComponentByChannel: dbmgr : 4000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=31980261564, lastReceivedTime=31968841058, inactivityExceptionPeriod=134887628)
  INFO root [2019-09-14 15:22:47,101] - Shutdowner::onShutdownBegin: shutting down(period=0, tickPeriod=1)
S_INFO root [2019-09-14 15:22:47,101] - onLoginAppShutDown()
  INFO root [2019-09-14 15:22:47,101] - Shutdowner::onShutdown: shutting down
  INFO root [2019-09-14 15:22:47,102] - Shutdowner::onShutdownEnd: shutting down
  INFO root [2019-09-14 15:22:47,116] - Script::uninstall(): is successfully!
 DEBUG root [2019-09-14 15:22:47,116] - ThreadPool::destroy(): starting size 2.
  WARN root [2019-09-14 15:22:47,418] - ThreadPool::destroy(): waiting for thread(2)[0x214b018,0x214b9d8,], try=1
  INFO root [2019-09-14 15:22:48,786] - 
  INFO root [2019-09-14 15:22:48,789] - -----------------------------------------------------------------------------------------


  INFO root [2019-09-14 15:22:48,790] - Resmgr::initialize: KBE_ROOT=D:/kbengine/
  INFO root [2019-09-14 15:22:48,790] - Resmgr::initialize: KBE_RES_PATH=D:/kbengine/kbe/res/;D:/kbengine/BattleOfBalls_assets/;D:/kbengine/BattleOfBalls_assets/scripts/;D:/kbengine/BattleOfBalls_assets/res/
  INFO root [2019-09-14 15:22:48,790] - Resmgr::initialize: KBE_BIN_PATH=D:/kbengine/kbe/bin/server/
  INFO root [2019-09-14 15:22:48,809] - NetworkInterface::initialize(EXTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:20013 )
  INFO root [2019-09-14 15:22:48,810] - NetworkInterface::initialize(EXTERNAL-TCP): address 192.168.56.1:20013, SOMAXCONN=128.
  INFO root [2019-09-14 15:22:48,814] - NetworkInterface::initialize(INTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:51366 )
  INFO root [2019-09-14 15:22:48,821] - NetworkInterface::initialize(INTERNAL-TCP): address 192.168.56.1:51366, SOMAXCONN=128.
  INFO root [2019-09-14 15:22:48,821] - server-configs:
  INFO root [2019-09-14 15:22:48,821] - 	internalTcpAddr : 192.168.56.1:51366
  INFO root [2019-09-14 15:22:48,822] - 	externalTcpAddr : 192.168.56.1:20013
  INFO root [2019-09-14 15:22:48,822] - 	componentID : 9000
  WARN root [2019-09-14 15:22:49,117] - invalid UID(0) <= 0, please check UID for environment! automatically set to 2082.
  WARN root [2019-09-14 15:22:49,542] - BundleBroadcast::BundleBroadcast: Cannot bind to port 20390, errorno=10048
  INFO root [2019-09-14 15:22:49,574] - ---- loginapp Version: 2.4.5. ScriptVersion: 0.1.0. Pythoncore: 3.7.3. Protocol: CE9FF53F4E85387EE4455C7191269324. Config: Debug X86. Built: 09:58:16 May  8 2019. AppID: 9000. UID: 2082. PID: 7144 ----
  INFO root [2019-09-14 15:22:49,575] - Server error_name: Intel(R) Core(TM) i5-6300HQ CPU @ 2.30GHz with 2097151kB RAM
  INFO root [2019-09-14 15:22:49,576] - ThreadPool::createThreadPool: creating  threadpool...
  INFO root [2019-09-14 15:22:49,576] - ThreadPool::createThreadPool: successfully(2), newThreadCount=1, normalMaxThreadCount=2, maxThreadCount=8
  INFO root [2019-09-14 15:22:49,823] - Script::install(): is successfully, Python=(3.7.3 (default, May  8 2019, 09:53:38) [MSC v.1916 32 bit (Intel)])!
  INFO root [2019-09-14 15:22:49,823] - Script::install(): KBExtra is successfully!
S_INFO root [2019-09-14 15:22:49,839] - onLoginAppReady: bootstrapGroupIndex=0, bootstrapGlobalIndex=0
  INFO root [2019-09-14 15:22:49,839] - TelnetServer server is running on port 31000
  INFO root [2019-09-14 15:22:50,282] - ---- loginapp is running ----
 DEBUG root [2019-09-14 15:22:50,283] - Components::process(): Request for the process of identity...
  INFO root [2019-09-14 15:22:57,499] - Components::findComponents: find logger(1)...
  INFO root [2019-09-14 15:22:57,595] - Components::findComponents: found logger, addr:192.168.56.1:51368
  INFO root [2019-09-14 15:22:57,595] - Components::addComponent: reset orderLog, uid=2082!
  INFO root [2019-09-14 15:22:57,595] - Components::addComponent[logger], uid=2082, componentID=2000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2019-09-14 15:22:57,596] - Components::findComponents: find dbmgr(2)...
  INFO root [2019-09-14 15:22:57,601] - Components::findComponents: found dbmgr, addr:192.168.56.1:51369
  INFO root [2019-09-14 15:22:57,601] - Components::addComponent: reset orderLog, uid=2082!
  INFO root [2019-09-14 15:22:57,602] - Components::addComponent[dbmgr], uid=2082, componentID=4000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2019-09-14 15:22:57,605] - Components::findComponents: found logger, addr:192.168.56.1:51368
  INFO root [2019-09-14 15:22:57,610] - Components::findComponents: found dbmgr, addr:192.168.56.1:51369
  WARN root [2019-09-14 15:22:58,611] - Forwarding logs to logger[192.168.56.1:51368/0/0/0]...
 ERROR root [2019-09-14 15:24:32,436] - Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=32217299893, lastReceivedTime=32186702470, inactivityExceptionPeriod=134887564)
  INFO root [2019-09-14 15:24:32,438] - The following logs sent to logger failed:
  WARN root [2019-09-14 15:24:32,439] - ==> [2019-09-14 15:24:32 434] TCPPacketReceiver::processPendingEvents(192.168.56.1:51368): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-14 15:24:32,439] - ==> [2019-09-14 15:24:32 438] Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=32217299893, lastReceivedTime=32186702470, inactivityExceptionPeriod=134887564)
  WARN root [2019-09-14 15:24:32,536] - TCPPacketReceiver::processPendingEvents(192.168.56.1:51369): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-14 15:24:32,537] - Components::removeComponentByChannel: dbmgr : 4000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=32217526539, lastReceivedTime=32186725180, inactivityExceptionPeriod=134887564)
  INFO root [2019-09-14 15:24:32,545] - Shutdowner::onShutdownBegin: shutting down(period=0, tickPeriod=1)
S_INFO root [2019-09-14 15:24:32,548] - onLoginAppShutDown()
  INFO root [2019-09-14 15:24:32,549] - Shutdowner::onShutdown: shutting down
  INFO root [2019-09-14 15:24:32,549] - Shutdowner::onShutdownEnd: shutting down
  INFO root [2019-09-14 15:24:32,567] - Script::uninstall(): is successfully!
 DEBUG root [2019-09-14 15:24:32,567] - ThreadPool::destroy(): starting size 2.
  WARN root [2019-09-14 15:24:32,868] - ThreadPool::destroy(): waiting for thread(2)[0x1d5d250,0x1d5d0b0,], try=1
  INFO root [2019-09-14 15:29:31,273] - 
  INFO root [2019-09-14 15:29:31,276] - -----------------------------------------------------------------------------------------


  INFO root [2019-09-14 15:29:31,281] - Resmgr::initialize: KBE_ROOT=D:/kbengine/
  INFO root [2019-09-14 15:29:31,281] - Resmgr::initialize: KBE_RES_PATH=D:/kbengine/kbe/res/;D:/kbengine/BattleOfBalls_assets/;D:/kbengine/BattleOfBalls_assets/scripts/;D:/kbengine/BattleOfBalls_assets/res/
  INFO root [2019-09-14 15:29:31,281] - Resmgr::initialize: KBE_BIN_PATH=D:/kbengine/kbe/bin/server/
  INFO root [2019-09-14 15:29:31,350] - NetworkInterface::initialize(EXTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:20013 )
  INFO root [2019-09-14 15:29:31,350] - NetworkInterface::initialize(EXTERNAL-TCP): address 192.168.56.1:20013, SOMAXCONN=128.
  INFO root [2019-09-14 15:29:31,355] - NetworkInterface::initialize(INTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:51407 )
  INFO root [2019-09-14 15:29:31,355] - NetworkInterface::initialize(INTERNAL-TCP): address 192.168.56.1:51407, SOMAXCONN=128.
  INFO root [2019-09-14 15:29:31,357] - server-configs:
  INFO root [2019-09-14 15:29:31,357] - 	internalTcpAddr : 192.168.56.1:51407
  INFO root [2019-09-14 15:29:31,357] - 	externalTcpAddr : 192.168.56.1:20013
  INFO root [2019-09-14 15:29:31,357] - 	componentID : 9000
  WARN root [2019-09-14 15:29:31,667] - invalid UID(0) <= 0, please check UID for environment! automatically set to 2082.
  WARN root [2019-09-14 15:29:32,025] - BundleBroadcast::BundleBroadcast: Cannot bind to port 20166, errorno=10048
  INFO root [2019-09-14 15:29:32,032] - ---- loginapp Version: 2.4.5. ScriptVersion: 0.1.0. Pythoncore: 3.7.3. Protocol: CE9FF53F4E85387EE4455C7191269324. Config: Debug X86. Built: 09:58:16 May  8 2019. AppID: 9000. UID: 2082. PID: 13356 ----
  INFO root [2019-09-14 15:29:32,033] - Server error_name: Intel(R) Core(TM) i5-6300HQ CPU @ 2.30GHz with 2097151kB RAM
  INFO root [2019-09-14 15:29:32,042] - ThreadPool::createThreadPool: creating  threadpool...
  INFO root [2019-09-14 15:29:32,045] - ThreadPool::createThreadPool: successfully(2), newThreadCount=1, normalMaxThreadCount=2, maxThreadCount=8
  INFO root [2019-09-14 15:29:32,265] - Script::install(): is successfully, Python=(3.7.3 (default, May  8 2019, 09:53:38) [MSC v.1916 32 bit (Intel)])!
  INFO root [2019-09-14 15:29:32,266] - Script::install(): KBExtra is successfully!
S_INFO root [2019-09-14 15:29:32,281] - onLoginAppReady: bootstrapGroupIndex=0, bootstrapGlobalIndex=0
  INFO root [2019-09-14 15:29:32,282] - TelnetServer server is running on port 31000
  INFO root [2019-09-14 15:29:32,773] - ---- loginapp is running ----
 DEBUG root [2019-09-14 15:29:32,773] - Components::process(): Request for the process of identity...
  INFO root [2019-09-14 15:29:39,942] - Components::findComponents: find logger(1)...
  INFO root [2019-09-14 15:29:39,985] - Components::findComponents: found logger, addr:192.168.56.1:51404
  INFO root [2019-09-14 15:29:39,985] - Components::addComponent: reset orderLog, uid=2082!
  INFO root [2019-09-14 15:29:39,985] - Components::addComponent[logger], uid=2082, componentID=2000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2019-09-14 15:29:39,988] - Components::findComponents: find dbmgr(2)...
  INFO root [2019-09-14 15:29:39,995] - Components::findComponents: found dbmgr, addr:192.168.56.1:51411
  INFO root [2019-09-14 15:29:39,995] - Components::addComponent: reset orderLog, uid=2082!
  INFO root [2019-09-14 15:29:39,995] - Components::addComponent[dbmgr], uid=2082, componentID=4000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2019-09-14 15:29:40,098] - Components::findComponents: found dbmgr, addr:192.168.56.1:51411
  WARN root [2019-09-14 15:29:41,098] - Forwarding logs to logger[192.168.56.1:51404/0/0/0]...
 ERROR root [2019-09-14 15:31:24,775] - Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=33145061009, lastReceivedTime=33092327461, inactivityExceptionPeriod=134887564)
  INFO root [2019-09-14 15:31:24,781] - The following logs sent to logger failed:
  WARN root [2019-09-14 15:31:24,782] - ==> [2019-09-14 15:31:24 775] TCPPacketReceiver::processPendingEvents(192.168.56.1:51404): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-14 15:31:24,782] - ==> [2019-09-14 15:31:24 776] Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=33145061009, lastReceivedTime=33092327461, inactivityExceptionPeriod=134887564)
  WARN root [2019-09-14 15:31:24,882] - TCPPacketReceiver::processPendingEvents(192.168.56.1:51411): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-14 15:31:24,885] - Components::removeComponentByChannel: dbmgr : 4000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=33145308457, lastReceivedTime=33092602052, inactivityExceptionPeriod=134887564)
  INFO root [2019-09-14 15:31:24,906] - Shutdowner::onShutdownBegin: shutting down(period=0, tickPeriod=1)
S_INFO root [2019-09-14 15:31:24,913] - onLoginAppShutDown()
  INFO root [2019-09-14 15:31:24,914] - Shutdowner::onShutdown: shutting down
  INFO root [2019-09-14 15:31:24,914] - Shutdowner::onShutdownEnd: shutting down
  INFO root [2019-09-14 15:31:24,930] - Script::uninstall(): is successfully!
 DEBUG root [2019-09-14 15:31:24,931] - ThreadPool::destroy(): starting size 2.
  WARN root [2019-09-14 15:31:25,232] - ThreadPool::destroy(): waiting for thread(2)[0x1f6cad8,0x1f6cb40,], try=1
  INFO root [2019-09-14 15:34:29,739] - 
  INFO root [2019-09-14 15:34:29,747] - -----------------------------------------------------------------------------------------


  INFO root [2019-09-14 15:34:29,748] - Resmgr::initialize: KBE_ROOT=D:/kbengine/
  INFO root [2019-09-14 15:34:29,748] - Resmgr::initialize: KBE_RES_PATH=D:/kbengine/kbe/res/;D:/kbengine/BattleOfBalls_assets/;D:/kbengine/BattleOfBalls_assets/scripts/;D:/kbengine/BattleOfBalls_assets/res/
  INFO root [2019-09-14 15:34:29,748] - Resmgr::initialize: KBE_BIN_PATH=D:/kbengine/kbe/bin/server/
  INFO root [2019-09-14 15:34:29,779] - NetworkInterface::initialize(EXTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:20013 )
  INFO root [2019-09-14 15:34:29,779] - NetworkInterface::initialize(EXTERNAL-TCP): address 192.168.56.1:20013, SOMAXCONN=128.
  INFO root [2019-09-14 15:34:29,783] - NetworkInterface::initialize(INTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:51469 )
  INFO root [2019-09-14 15:34:29,783] - NetworkInterface::initialize(INTERNAL-TCP): address 192.168.56.1:51469, SOMAXCONN=128.
  INFO root [2019-09-14 15:34:29,783] - server-configs:
  INFO root [2019-09-14 15:34:29,783] - 	internalTcpAddr : 192.168.56.1:51469
  INFO root [2019-09-14 15:34:29,783] - 	externalTcpAddr : 192.168.56.1:20013
  INFO root [2019-09-14 15:34:29,783] - 	componentID : 9000
  WARN root [2019-09-14 15:34:29,990] - invalid UID(0) <= 0, please check UID for environment! automatically set to 2082.
  INFO root [2019-09-14 15:34:30,256] - Components::findLogger: found logger, addr:192.168.56.1:51466
  INFO root [2019-09-14 15:34:30,256] - Components::addComponent: reset orderLog, uid=2082!
  INFO root [2019-09-14 15:34:30,256] - Components::addComponent[logger], uid=2082, componentID=2000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2019-09-14 15:34:30,261] - ---- loginapp Version: 2.4.5. ScriptVersion: 0.1.0. Pythoncore: 3.7.3. Protocol: CE9FF53F4E85387EE4455C7191269324. Config: Debug X86. Built: 09:58:16 May  8 2019. AppID: 9000. UID: 2082. PID: 12688 ----
  INFO root [2019-09-14 15:34:30,261] - Server error_name: Intel(R) Core(TM) i5-6300HQ CPU @ 2.30GHz with 2097151kB RAM
  INFO root [2019-09-14 15:34:30,262] - ThreadPool::createThreadPool: creating  threadpool...
  INFO root [2019-09-14 15:34:30,262] - ThreadPool::createThreadPool: successfully(2), newThreadCount=1, normalMaxThreadCount=2, maxThreadCount=8
  INFO root [2019-09-14 15:34:30,461] - Script::install(): is successfully, Python=(3.7.3 (default, May  8 2019, 09:53:38) [MSC v.1916 32 bit (Intel)])!
  INFO root [2019-09-14 15:34:30,462] - Script::install(): KBExtra is successfully!
S_INFO root [2019-09-14 15:34:30,473] - onLoginAppReady: bootstrapGroupIndex=0, bootstrapGlobalIndex=0
  INFO root [2019-09-14 15:34:30,474] - TelnetServer server is running on port 31000
  INFO root [2019-09-14 15:34:30,859] - ---- loginapp is running ----
 DEBUG root [2019-09-14 15:34:30,859] - Components::process(): Request for the process of identity...
  WARN root [2019-09-14 15:34:36,970] - Forwarding logs to logger[192.168.56.1:51466/0/0/0]...
 ERROR root [2019-09-14 15:36:56,952] - Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=33892460709, lastReceivedTime=33831255153, inactivityExceptionPeriod=134887437)
  INFO root [2019-09-14 15:36:56,956] - The following logs sent to logger failed:
  WARN root [2019-09-14 15:36:56,956] - ==> [2019-09-14 15:36:56 950] TCPPacketReceiver::processPendingEvents(192.168.56.1:51466): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-14 15:36:56,956] - ==> [2019-09-14 15:36:56 953] Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=33892460709, lastReceivedTime=33831255153, inactivityExceptionPeriod=134887437)
  WARN root [2019-09-14 15:36:57,066] - TCPPacketReceiver::processPendingEvents(192.168.56.1:51471): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-14 15:36:57,067] - Components::removeComponentByChannel: dbmgr : 4000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=33892719934, lastReceivedTime=33831346253, inactivityExceptionPeriod=134887437)
  INFO root [2019-09-14 15:36:57,071] - Shutdowner::onShutdownBegin: shutting down(period=0, tickPeriod=1)
S_INFO root [2019-09-14 15:36:57,072] - onLoginAppShutDown()
  INFO root [2019-09-14 15:36:57,072] - Shutdowner::onShutdown: shutting down
  INFO root [2019-09-14 15:36:57,072] - Shutdowner::onShutdownEnd: shutting down
  INFO root [2019-09-14 15:36:57,088] - Script::uninstall(): is successfully!
 DEBUG root [2019-09-14 15:36:57,088] - ThreadPool::destroy(): starting size 2.
  WARN root [2019-09-14 15:36:57,389] - ThreadPool::destroy(): waiting for thread(2)[0x892a90,0x892af8,], try=1
  INFO root [2019-09-14 15:40:16,180] - 
  INFO root [2019-09-14 15:40:16,182] - -----------------------------------------------------------------------------------------


  INFO root [2019-09-14 15:40:16,183] - Resmgr::initialize: KBE_ROOT=D:/kbengine/
  INFO root [2019-09-14 15:40:16,183] - Resmgr::initialize: KBE_RES_PATH=D:/kbengine/kbe/res/;D:/kbengine/BattleOfBalls_assets/;D:/kbengine/BattleOfBalls_assets/scripts/;D:/kbengine/BattleOfBalls_assets/res/
  INFO root [2019-09-14 15:40:16,184] - Resmgr::initialize: KBE_BIN_PATH=D:/kbengine/kbe/bin/server/
  INFO root [2019-09-14 15:40:16,207] - NetworkInterface::initialize(EXTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:20013 )
  INFO root [2019-09-14 15:40:16,207] - NetworkInterface::initialize(EXTERNAL-TCP): address 192.168.56.1:20013, SOMAXCONN=128.
  INFO root [2019-09-14 15:40:16,212] - NetworkInterface::initialize(INTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:51510 )
  INFO root [2019-09-14 15:40:16,213] - NetworkInterface::initialize(INTERNAL-TCP): address 192.168.56.1:51510, SOMAXCONN=128.
  INFO root [2019-09-14 15:40:16,213] - server-configs:
  INFO root [2019-09-14 15:40:16,213] - 	internalTcpAddr : 192.168.56.1:51510
  INFO root [2019-09-14 15:40:16,213] - 	externalTcpAddr : 192.168.56.1:20013
  INFO root [2019-09-14 15:40:16,213] - 	componentID : 9000
  WARN root [2019-09-14 15:40:16,422] - invalid UID(0) <= 0, please check UID for environment! automatically set to 2082.
  INFO root [2019-09-14 15:40:16,853] - Components::findLogger: found logger, addr:192.168.56.1:51504
  INFO root [2019-09-14 15:40:16,853] - Components::addComponent: reset orderLog, uid=2082!
  INFO root [2019-09-14 15:40:16,853] - Components::addComponent[logger], uid=2082, componentID=2000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2019-09-14 15:40:16,860] - ---- loginapp Version: 2.4.5. ScriptVersion: 0.1.0. Pythoncore: 3.7.3. Protocol: CE9FF53F4E85387EE4455C7191269324. Config: Debug X86. Built: 09:58:16 May  8 2019. AppID: 9000. UID: 2082. PID: 12128 ----
  INFO root [2019-09-14 15:40:16,860] - Server error_name: Intel(R) Core(TM) i5-6300HQ CPU @ 2.30GHz with 2097151kB RAM
  INFO root [2019-09-14 15:40:16,860] - ThreadPool::createThreadPool: creating  threadpool...
  INFO root [2019-09-14 15:40:16,861] - ThreadPool::createThreadPool: successfully(2), newThreadCount=1, normalMaxThreadCount=2, maxThreadCount=8
  INFO root [2019-09-14 15:40:17,059] - Script::install(): is successfully, Python=(3.7.3 (default, May  8 2019, 09:53:38) [MSC v.1916 32 bit (Intel)])!
  INFO root [2019-09-14 15:40:17,059] - Script::install(): KBExtra is successfully!
S_INFO root [2019-09-14 15:40:17,076] - onLoginAppReady: bootstrapGroupIndex=0, bootstrapGlobalIndex=0
  INFO root [2019-09-14 15:40:17,078] - TelnetServer server is running on port 31000
  INFO root [2019-09-14 15:40:17,505] - ---- loginapp is running ----
 DEBUG root [2019-09-14 15:40:17,506] - Components::process(): Request for the process of identity...
  WARN root [2019-09-14 15:40:23,636] - Forwarding logs to logger[192.168.56.1:51504/0/0/0]...
 ERROR root [2019-09-14 15:43:14,746] - Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=34742498341, lastReceivedTime=34678092376, inactivityExceptionPeriod=134887564)
  INFO root [2019-09-14 15:43:14,750] - The following logs sent to logger failed:
  WARN root [2019-09-14 15:43:14,750] - ==> [2019-09-14 15:43:14 742] TCPPacketReceiver::processPendingEvents(192.168.56.1:51504): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-14 15:43:14,751] - ==> [2019-09-14 15:43:14 748] Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=34742498341, lastReceivedTime=34678092376, inactivityExceptionPeriod=134887564)
  WARN root [2019-09-14 15:43:14,842] - TCPPacketReceiver::processPendingEvents(192.168.56.1:51507): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-14 15:43:14,848] - Components::removeComponentByChannel: dbmgr : 4000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=34742725302, lastReceivedTime=34678192655, inactivityExceptionPeriod=134887564)
  INFO root [2019-09-14 15:43:14,862] - Shutdowner::onShutdownBegin: shutting down(period=0, tickPeriod=1)
S_INFO root [2019-09-14 15:43:14,871] - onLoginAppShutDown()
  INFO root [2019-09-14 15:43:14,871] - Shutdowner::onShutdown: shutting down
  INFO root [2019-09-14 15:43:14,871] - Shutdowner::onShutdownEnd: shutting down
  INFO root [2019-09-14 15:43:14,882] - Script::uninstall(): is successfully!
 DEBUG root [2019-09-14 15:43:14,882] - ThreadPool::destroy(): starting size 2.
  WARN root [2019-09-14 15:43:15,183] - ThreadPool::destroy(): waiting for thread(2)[0x7434e0,0x743a90,], try=1
  INFO root [2019-09-14 15:43:16,649] - 
  INFO root [2019-09-14 15:43:16,655] - -----------------------------------------------------------------------------------------


  INFO root [2019-09-14 15:43:16,656] - Resmgr::initialize: KBE_ROOT=D:/kbengine/
  INFO root [2019-09-14 15:43:16,656] - Resmgr::initialize: KBE_RES_PATH=D:/kbengine/kbe/res/;D:/kbengine/BattleOfBalls_assets/;D:/kbengine/BattleOfBalls_assets/scripts/;D:/kbengine/BattleOfBalls_assets/res/
  INFO root [2019-09-14 15:43:16,657] - Resmgr::initialize: KBE_BIN_PATH=D:/kbengine/kbe/bin/server/
  INFO root [2019-09-14 15:43:16,679] - NetworkInterface::initialize(EXTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:20013 )
  INFO root [2019-09-14 15:43:16,679] - NetworkInterface::initialize(EXTERNAL-TCP): address 192.168.56.1:20013, SOMAXCONN=128.
  INFO root [2019-09-14 15:43:16,683] - NetworkInterface::initialize(INTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:51551 )
  INFO root [2019-09-14 15:43:16,683] - NetworkInterface::initialize(INTERNAL-TCP): address 192.168.56.1:51551, SOMAXCONN=128.
  INFO root [2019-09-14 15:43:16,684] - server-configs:
  INFO root [2019-09-14 15:43:16,684] - 	internalTcpAddr : 192.168.56.1:51551
  INFO root [2019-09-14 15:43:16,684] - 	externalTcpAddr : 192.168.56.1:20013
  INFO root [2019-09-14 15:43:16,684] - 	componentID : 9000
  WARN root [2019-09-14 15:43:16,910] - invalid UID(0) <= 0, please check UID for environment! automatically set to 2082.
  INFO root [2019-09-14 15:43:17,065] - Components::findLogger: found logger, addr:192.168.56.1:51546
  INFO root [2019-09-14 15:43:17,066] - Components::addComponent: reset orderLog, uid=2082!
  INFO root [2019-09-14 15:43:17,066] - Components::addComponent[logger], uid=2082, componentID=2000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2019-09-14 15:43:17,071] - ---- loginapp Version: 2.4.5. ScriptVersion: 0.1.0. Pythoncore: 3.7.3. Protocol: CE9FF53F4E85387EE4455C7191269324. Config: Debug X86. Built: 09:58:16 May  8 2019. AppID: 9000. UID: 2082. PID: 10400 ----
  INFO root [2019-09-14 15:43:17,071] - Server error_name: Intel(R) Core(TM) i5-6300HQ CPU @ 2.30GHz with 2097151kB RAM
  INFO root [2019-09-14 15:43:17,072] - ThreadPool::createThreadPool: creating  threadpool...
  INFO root [2019-09-14 15:43:17,072] - ThreadPool::createThreadPool: successfully(2), newThreadCount=1, normalMaxThreadCount=2, maxThreadCount=8
  INFO root [2019-09-14 15:43:17,273] - Script::install(): is successfully, Python=(3.7.3 (default, May  8 2019, 09:53:38) [MSC v.1916 32 bit (Intel)])!
  INFO root [2019-09-14 15:43:17,273] - Script::install(): KBExtra is successfully!
S_INFO root [2019-09-14 15:43:17,285] - onLoginAppReady: bootstrapGroupIndex=0, bootstrapGlobalIndex=0
  INFO root [2019-09-14 15:43:17,286] - TelnetServer server is running on port 31000
  INFO root [2019-09-14 15:43:17,694] - ---- loginapp is running ----
 DEBUG root [2019-09-14 15:43:17,695] - Components::process(): Request for the process of identity...
  WARN root [2019-09-14 15:43:23,806] - Forwarding logs to logger[192.168.56.1:51546/0/0/0]...
 ERROR root [2019-09-14 15:47:41,000] - Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=35341567362, lastReceivedTime=35286497882, inactivityExceptionPeriod=134887564)
  INFO root [2019-09-14 15:47:41,005] - The following logs sent to logger failed:
  WARN root [2019-09-14 15:47:41,006] - ==> [2019-09-14 15:47:40 994] TCPPacketReceiver::processPendingEvents(192.168.56.1:51546): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-14 15:47:41,006] - ==> [2019-09-14 15:47:41 000] Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=35341567362, lastReceivedTime=35286497882, inactivityExceptionPeriod=134887564)
  WARN root [2019-09-14 15:47:41,123] - TCPPacketReceiver::processPendingEvents(192.168.56.1:51547): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-14 15:47:41,140] - Components::removeComponentByChannel: dbmgr : 4000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=35341882692, lastReceivedTime=35286533500, inactivityExceptionPeriod=134887564)
  INFO root [2019-09-14 15:47:41,143] - Shutdowner::onShutdownBegin: shutting down(period=0, tickPeriod=1)
S_INFO root [2019-09-14 15:47:41,144] - onLoginAppShutDown()
  INFO root [2019-09-14 15:47:41,144] - Shutdowner::onShutdown: shutting down
  INFO root [2019-09-14 15:47:41,144] - Shutdowner::onShutdownEnd: shutting down
  INFO root [2019-09-14 15:47:41,163] - Script::uninstall(): is successfully!
 DEBUG root [2019-09-14 15:47:41,164] - ThreadPool::destroy(): starting size 2.
  WARN root [2019-09-14 15:47:41,465] - ThreadPool::destroy(): waiting for thread(2)[0x1e104e0,0x1e0fb88,], try=1
  INFO root [2019-09-14 15:47:46,455] - 
  INFO root [2019-09-14 15:47:46,456] - -----------------------------------------------------------------------------------------


  INFO root [2019-09-14 15:47:46,457] - Resmgr::initialize: KBE_ROOT=D:/kbengine/
  INFO root [2019-09-14 15:47:46,457] - Resmgr::initialize: KBE_RES_PATH=D:/kbengine/kbe/res/;D:/kbengine/BattleOfBalls_assets/;D:/kbengine/BattleOfBalls_assets/scripts/;D:/kbengine/BattleOfBalls_assets/res/
  INFO root [2019-09-14 15:47:46,458] - Resmgr::initialize: KBE_BIN_PATH=D:/kbengine/kbe/bin/server/
  INFO root [2019-09-14 15:47:46,499] - NetworkInterface::initialize(EXTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:20013 )
  INFO root [2019-09-14 15:47:46,500] - NetworkInterface::initialize(EXTERNAL-TCP): address 192.168.56.1:20013, SOMAXCONN=128.
  INFO root [2019-09-14 15:47:46,508] - NetworkInterface::initialize(INTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:51589 )
  INFO root [2019-09-14 15:47:46,509] - NetworkInterface::initialize(INTERNAL-TCP): address 192.168.56.1:51589, SOMAXCONN=128.
  INFO root [2019-09-14 15:47:46,509] - server-configs:
  INFO root [2019-09-14 15:47:46,509] - 	internalTcpAddr : 192.168.56.1:51589
  INFO root [2019-09-14 15:47:46,509] - 	externalTcpAddr : 192.168.56.1:20013
  INFO root [2019-09-14 15:47:46,509] - 	componentID : 9000
  WARN root [2019-09-14 15:47:46,783] - invalid UID(0) <= 0, please check UID for environment! automatically set to 2082.
  INFO root [2019-09-14 15:47:47,077] - Components::findLogger: found logger, addr:192.168.56.1:51583
  INFO root [2019-09-14 15:47:47,077] - Components::addComponent: reset orderLog, uid=2082!
  INFO root [2019-09-14 15:47:47,078] - Components::addComponent[logger], uid=2082, componentID=2000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2019-09-14 15:47:47,088] - ---- loginapp Version: 2.4.5. ScriptVersion: 0.1.0. Pythoncore: 3.7.3. Protocol: CE9FF53F4E85387EE4455C7191269324. Config: Debug X86. Built: 09:58:16 May  8 2019. AppID: 9000. UID: 2082. PID: 9880 ----
  INFO root [2019-09-14 15:47:47,089] - Server error_name: Intel(R) Core(TM) i5-6300HQ CPU @ 2.30GHz with 2097151kB RAM
  INFO root [2019-09-14 15:47:47,090] - ThreadPool::createThreadPool: creating  threadpool...
  INFO root [2019-09-14 15:47:47,090] - ThreadPool::createThreadPool: successfully(2), newThreadCount=1, normalMaxThreadCount=2, maxThreadCount=8
  INFO root [2019-09-14 15:47:47,307] - Script::install(): is successfully, Python=(3.7.3 (default, May  8 2019, 09:53:38) [MSC v.1916 32 bit (Intel)])!
  INFO root [2019-09-14 15:47:47,307] - Script::install(): KBExtra is successfully!
S_INFO root [2019-09-14 15:47:47,320] - onLoginAppReady: bootstrapGroupIndex=0, bootstrapGlobalIndex=0
  INFO root [2019-09-14 15:47:47,321] - TelnetServer server is running on port 31000
  INFO root [2019-09-14 15:47:47,811] - ---- loginapp is running ----
 DEBUG root [2019-09-14 15:47:47,812] - Components::process(): Request for the process of identity...
  WARN root [2019-09-14 15:47:53,947] - Forwarding logs to logger[192.168.56.1:51583/0/0/0]...
 ERROR root [2019-09-14 15:51:01,900] - Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=35793593598, lastReceivedTime=35758624085, inactivityExceptionPeriod=134887564)
  INFO root [2019-09-14 15:51:01,905] - The following logs sent to logger failed:
  WARN root [2019-09-14 15:51:01,905] - ==> [2019-09-14 15:51:01 899] TCPPacketReceiver::processPendingEvents(192.168.56.1:51583): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-14 15:51:01,905] - ==> [2019-09-14 15:51:01 901] Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=35793593598, lastReceivedTime=35758624085, inactivityExceptionPeriod=134887564)
  WARN root [2019-09-14 15:51:02,017] - TCPPacketReceiver::processPendingEvents(192.168.56.1:51586): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-14 15:51:02,021] - Components::removeComponentByChannel: dbmgr : 4000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=35793865275, lastReceivedTime=35758850006, inactivityExceptionPeriod=134887564)
  INFO root [2019-09-14 15:51:02,026] - Shutdowner::onShutdownBegin: shutting down(period=0, tickPeriod=1)
S_INFO root [2019-09-14 15:51:02,028] - onLoginAppShutDown()
  INFO root [2019-09-14 15:51:02,028] - Shutdowner::onShutdown: shutting down
  INFO root [2019-09-14 15:51:02,028] - Shutdowner::onShutdownEnd: shutting down
  INFO root [2019-09-14 15:51:02,042] - Script::uninstall(): is successfully!
 DEBUG root [2019-09-14 15:51:02,043] - ThreadPool::destroy(): starting size 2.
  WARN root [2019-09-14 15:51:02,345] - ThreadPool::destroy(): waiting for thread(2)[0x203f88,0x203f20,], try=1
  INFO root [2019-09-14 15:54:13,326] - 
  INFO root [2019-09-14 15:54:13,331] - -----------------------------------------------------------------------------------------


  INFO root [2019-09-14 15:54:13,332] - Resmgr::initialize: KBE_ROOT=D:/kbengine/
  INFO root [2019-09-14 15:54:13,332] - Resmgr::initialize: KBE_RES_PATH=D:/kbengine/kbe/res/;D:/kbengine/BattleOfBalls_assets/;D:/kbengine/BattleOfBalls_assets/scripts/;D:/kbengine/BattleOfBalls_assets/res/
  INFO root [2019-09-14 15:54:13,332] - Resmgr::initialize: KBE_BIN_PATH=D:/kbengine/kbe/bin/server/
  INFO root [2019-09-14 15:54:13,375] - NetworkInterface::initialize(EXTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:20013 )
  INFO root [2019-09-14 15:54:13,375] - NetworkInterface::initialize(EXTERNAL-TCP): address 192.168.56.1:20013, SOMAXCONN=128.
  INFO root [2019-09-14 15:54:13,384] - NetworkInterface::initialize(INTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:51628 )
  INFO root [2019-09-14 15:54:13,384] - NetworkInterface::initialize(INTERNAL-TCP): address 192.168.56.1:51628, SOMAXCONN=128.
  INFO root [2019-09-14 15:54:13,384] - server-configs:
  INFO root [2019-09-14 15:54:13,384] - 	internalTcpAddr : 192.168.56.1:51628
  INFO root [2019-09-14 15:54:13,384] - 	externalTcpAddr : 192.168.56.1:20013
  INFO root [2019-09-14 15:54:13,384] - 	componentID : 9000
  WARN root [2019-09-14 15:54:13,600] - invalid UID(0) <= 0, please check UID for environment! automatically set to 2082.
  INFO root [2019-09-14 15:54:15,104] - ---- loginapp Version: 2.4.5. ScriptVersion: 0.1.0. Pythoncore: 3.7.3. Protocol: CE9FF53F4E85387EE4455C7191269324. Config: Debug X86. Built: 09:58:16 May  8 2019. AppID: 9000. UID: 2082. PID: 6576 ----
  INFO root [2019-09-14 15:54:15,105] - Server error_name: Intel(R) Core(TM) i5-6300HQ CPU @ 2.30GHz with 2097151kB RAM
  INFO root [2019-09-14 15:54:15,105] - ThreadPool::createThreadPool: creating  threadpool...
  INFO root [2019-09-14 15:54:15,105] - ThreadPool::createThreadPool: successfully(2), newThreadCount=1, normalMaxThreadCount=2, maxThreadCount=8
  INFO root [2019-09-14 15:54:15,338] - Script::install(): is successfully, Python=(3.7.3 (default, May  8 2019, 09:53:38) [MSC v.1916 32 bit (Intel)])!
  INFO root [2019-09-14 15:54:15,338] - Script::install(): KBExtra is successfully!
S_INFO root [2019-09-14 15:54:15,351] - onLoginAppReady: bootstrapGroupIndex=0, bootstrapGlobalIndex=0
  INFO root [2019-09-14 15:54:15,351] - TelnetServer server is running on port 31000
  INFO root [2019-09-14 15:54:16,032] - ---- loginapp is running ----
 DEBUG root [2019-09-14 15:54:16,034] - Components::process(): Request for the process of identity...
  INFO root [2019-09-14 15:54:23,213] - Components::findComponents: find logger(1)...
  INFO root [2019-09-14 15:54:23,263] - Components::findComponents: found logger, addr:192.168.56.1:51626
  INFO root [2019-09-14 15:54:23,263] - Components::addComponent: reset orderLog, uid=2082!
  INFO root [2019-09-14 15:54:23,264] - Components::addComponent[logger], uid=2082, componentID=2000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2019-09-14 15:54:23,267] - Components::findComponents: find dbmgr(2)...
  INFO root [2019-09-14 15:54:23,372] - Components::findComponents: found dbmgr, addr:192.168.56.1:51623
  INFO root [2019-09-14 15:54:23,372] - Components::addComponent: reset orderLog, uid=2082!
  INFO root [2019-09-14 15:54:23,372] - Components::addComponent[dbmgr], uid=2082, componentID=4000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2019-09-14 15:54:23,378] - Components::findComponents: found dbmgr, addr:192.168.56.1:51623
  WARN root [2019-09-14 15:54:24,379] - Forwarding logs to logger[192.168.56.1:51626/0/0/0]...
 ERROR root [2019-09-14 15:58:58,771] - Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=36866553219, lastReceivedTime=36831901183, inactivityExceptionPeriod=134887564)
  INFO root [2019-09-14 15:58:58,774] - The following logs sent to logger failed:
  WARN root [2019-09-14 15:58:58,774] - ==> [2019-09-14 15:58:58 764] TCPPacketReceiver::processPendingEvents(192.168.56.1:51626): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-14 15:58:58,775] - ==> [2019-09-14 15:58:58 772] Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=36866553219, lastReceivedTime=36831901183, inactivityExceptionPeriod=134887564)
  WARN root [2019-09-14 15:58:58,888] - TCPPacketReceiver::processPendingEvents(192.168.56.1:51623): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-14 15:58:58,891] - Components::removeComponentByChannel: dbmgr : 4000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=36866823008, lastReceivedTime=36831858241, inactivityExceptionPeriod=134887564)
  INFO root [2019-09-14 15:58:58,909] - Shutdowner::onShutdownBegin: shutting down(period=0, tickPeriod=1)
S_INFO root [2019-09-14 15:58:58,912] - onLoginAppShutDown()
  INFO root [2019-09-14 15:58:58,912] - Shutdowner::onShutdown: shutting down
  INFO root [2019-09-14 15:58:58,912] - Shutdowner::onShutdownEnd: shutting down
  INFO root [2019-09-14 15:58:58,933] - Script::uninstall(): is successfully!
 DEBUG root [2019-09-14 15:58:58,933] - ThreadPool::destroy(): starting size 2.
  WARN root [2019-09-14 15:58:59,235] - ThreadPool::destroy(): waiting for thread(2)[0x56d0b0,0x56d590,], try=1
  INFO root [2019-09-14 16:00:37,105] - 
  INFO root [2019-09-14 16:00:37,105] - -----------------------------------------------------------------------------------------


  INFO root [2019-09-14 16:00:37,106] - Resmgr::initialize: KBE_ROOT=D:/kbengine/
  INFO root [2019-09-14 16:00:37,106] - Resmgr::initialize: KBE_RES_PATH=D:/kbengine/kbe/res/;D:/kbengine/BattleOfBalls_assets/;D:/kbengine/BattleOfBalls_assets/scripts/;D:/kbengine/BattleOfBalls_assets/res/
  INFO root [2019-09-14 16:00:37,106] - Resmgr::initialize: KBE_BIN_PATH=D:/kbengine/kbe/bin/server/
  INFO root [2019-09-14 16:00:37,125] - NetworkInterface::initialize(EXTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:20013 )
  INFO root [2019-09-14 16:00:37,125] - NetworkInterface::initialize(EXTERNAL-TCP): address 192.168.56.1:20013, SOMAXCONN=128.
  INFO root [2019-09-14 16:00:37,138] - NetworkInterface::initialize(INTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:51668 )
  INFO root [2019-09-14 16:00:37,139] - NetworkInterface::initialize(INTERNAL-TCP): address 192.168.56.1:51668, SOMAXCONN=128.
  INFO root [2019-09-14 16:00:37,139] - server-configs:
  INFO root [2019-09-14 16:00:37,140] - 	internalTcpAddr : 192.168.56.1:51668
  INFO root [2019-09-14 16:00:37,140] - 	externalTcpAddr : 192.168.56.1:20013
  INFO root [2019-09-14 16:00:37,140] - 	componentID : 9000
  WARN root [2019-09-14 16:00:37,429] - invalid UID(0) <= 0, please check UID for environment! automatically set to 2082.
  INFO root [2019-09-14 16:00:38,944] - ---- loginapp Version: 2.4.5. ScriptVersion: 0.1.0. Pythoncore: 3.7.3. Protocol: CE9FF53F4E85387EE4455C7191269324. Config: Debug X86. Built: 09:58:16 May  8 2019. AppID: 9000. UID: 2082. PID: 5292 ----
  INFO root [2019-09-14 16:00:38,945] - Server error_name: Intel(R) Core(TM) i5-6300HQ CPU @ 2.30GHz with 2097151kB RAM
  INFO root [2019-09-14 16:00:38,945] - ThreadPool::createThreadPool: creating  threadpool...
  INFO root [2019-09-14 16:00:38,945] - ThreadPool::createThreadPool: successfully(2), newThreadCount=1, normalMaxThreadCount=2, maxThreadCount=8
  INFO root [2019-09-14 16:00:39,223] - Script::install(): is successfully, Python=(3.7.3 (default, May  8 2019, 09:53:38) [MSC v.1916 32 bit (Intel)])!
  INFO root [2019-09-14 16:00:39,223] - Script::install(): KBExtra is successfully!
S_INFO root [2019-09-14 16:00:39,244] - onLoginAppReady: bootstrapGroupIndex=0, bootstrapGlobalIndex=0
  INFO root [2019-09-14 16:00:39,244] - TelnetServer server is running on port 31000
  INFO root [2019-09-14 16:00:39,766] - ---- loginapp is running ----
 DEBUG root [2019-09-14 16:00:39,767] - Components::process(): Request for the process of identity...
  INFO root [2019-09-14 16:00:47,005] - Components::findComponents: find logger(1)...
  INFO root [2019-09-14 16:00:47,062] - Components::findComponents: found logger, addr:192.168.56.1:51665
  INFO root [2019-09-14 16:00:47,063] - Components::addComponent: reset orderLog, uid=2082!
  INFO root [2019-09-14 16:00:47,063] - Components::addComponent[logger], uid=2082, componentID=2000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2019-09-14 16:00:47,066] - Components::findComponents: find dbmgr(2)...
  INFO root [2019-09-14 16:00:47,074] - Components::findComponents: found logger, addr:192.168.56.1:51665
  INFO root [2019-09-14 16:00:47,078] - Components::findComponents: found dbmgr, addr:192.168.56.1:51671
  INFO root [2019-09-14 16:00:47,078] - Components::addComponent: reset orderLog, uid=2082!
  INFO root [2019-09-14 16:00:47,078] - Components::addComponent[dbmgr], uid=2082, componentID=4000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2019-09-14 16:00:47,179] - Components::findComponents: found dbmgr, addr:192.168.56.1:51671
  WARN root [2019-09-14 16:00:48,180] - Forwarding logs to logger[192.168.56.1:51665/0/0/0]...
 ERROR root [2019-09-14 16:03:00,764] - Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=37411038673, lastReceivedTime=37357511692, inactivityExceptionPeriod=134887564)
  INFO root [2019-09-14 16:03:00,776] - The following logs sent to logger failed:
  WARN root [2019-09-14 16:03:00,776] - ==> [2019-09-14 16:03:00 760] TCPPacketReceiver::processPendingEvents(192.168.56.1:51665): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-14 16:03:00,776] - ==> [2019-09-14 16:03:00 765] Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=37411038673, lastReceivedTime=37357511692, inactivityExceptionPeriod=134887564)
  WARN root [2019-09-14 16:03:00,884] - TCPPacketReceiver::processPendingEvents(192.168.56.1:51671): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-14 16:03:00,894] - Components::removeComponentByChannel: dbmgr : 4000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=37411332135, lastReceivedTime=37358106395, inactivityExceptionPeriod=134887564)
  INFO root [2019-09-14 16:03:00,912] - Shutdowner::onShutdownBegin: shutting down(period=0, tickPeriod=1)
S_INFO root [2019-09-14 16:03:00,912] - onLoginAppShutDown()
  INFO root [2019-09-14 16:03:00,912] - Shutdowner::onShutdown: shutting down
  INFO root [2019-09-14 16:03:00,913] - Shutdowner::onShutdownEnd: shutting down
  INFO root [2019-09-14 16:03:00,930] - Script::uninstall(): is successfully!
 DEBUG root [2019-09-14 16:03:00,930] - ThreadPool::destroy(): starting size 2.
  WARN root [2019-09-14 16:03:01,232] - ThreadPool::destroy(): waiting for thread(2)[0x1ffd068,0x1ffc980,], try=1
  INFO root [2019-09-14 16:03:02,875] - 
  INFO root [2019-09-14 16:03:02,878] - -----------------------------------------------------------------------------------------


  INFO root [2019-09-14 16:03:02,879] - Resmgr::initialize: KBE_ROOT=D:/kbengine/
  INFO root [2019-09-14 16:03:02,879] - Resmgr::initialize: KBE_RES_PATH=D:/kbengine/kbe/res/;D:/kbengine/BattleOfBalls_assets/;D:/kbengine/BattleOfBalls_assets/scripts/;D:/kbengine/BattleOfBalls_assets/res/
  INFO root [2019-09-14 16:03:02,879] - Resmgr::initialize: KBE_BIN_PATH=D:/kbengine/kbe/bin/server/
  INFO root [2019-09-14 16:03:02,944] - NetworkInterface::initialize(EXTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:20013 )
  INFO root [2019-09-14 16:03:02,944] - NetworkInterface::initialize(EXTERNAL-TCP): address 192.168.56.1:20013, SOMAXCONN=128.
  INFO root [2019-09-14 16:03:02,956] - NetworkInterface::initialize(INTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:51710 )
  INFO root [2019-09-14 16:03:02,957] - NetworkInterface::initialize(INTERNAL-TCP): address 192.168.56.1:51710, SOMAXCONN=128.
  INFO root [2019-09-14 16:03:02,957] - server-configs:
  INFO root [2019-09-14 16:03:02,957] - 	internalTcpAddr : 192.168.56.1:51710
  INFO root [2019-09-14 16:03:02,957] - 	externalTcpAddr : 192.168.56.1:20013
  INFO root [2019-09-14 16:03:02,957] - 	componentID : 9000
  WARN root [2019-09-14 16:03:03,226] - invalid UID(0) <= 0, please check UID for environment! automatically set to 2082.
  INFO root [2019-09-14 16:03:03,685] - Components::findLogger: found logger, addr:192.168.56.1:51709
  INFO root [2019-09-14 16:03:03,685] - Components::addComponent: reset orderLog, uid=2082!
  INFO root [2019-09-14 16:03:03,686] - Components::addComponent[logger], uid=2082, componentID=2000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2019-09-14 16:03:03,692] - ---- loginapp Version: 2.4.5. ScriptVersion: 0.1.0. Pythoncore: 3.7.3. Protocol: CE9FF53F4E85387EE4455C7191269324. Config: Debug X86. Built: 09:58:16 May  8 2019. AppID: 9000. UID: 2082. PID: 7216 ----
  INFO root [2019-09-14 16:03:03,692] - Server error_name: Intel(R) Core(TM) i5-6300HQ CPU @ 2.30GHz with 2097151kB RAM
  INFO root [2019-09-14 16:03:03,692] - ThreadPool::createThreadPool: creating  threadpool...
  INFO root [2019-09-14 16:03:03,693] - ThreadPool::createThreadPool: successfully(2), newThreadCount=1, normalMaxThreadCount=2, maxThreadCount=8
  INFO root [2019-09-14 16:03:03,958] - Script::install(): is successfully, Python=(3.7.3 (default, May  8 2019, 09:53:38) [MSC v.1916 32 bit (Intel)])!
  INFO root [2019-09-14 16:03:03,958] - Script::install(): KBExtra is successfully!
S_INFO root [2019-09-14 16:03:04,007] - onLoginAppReady: bootstrapGroupIndex=0, bootstrapGlobalIndex=0
  INFO root [2019-09-14 16:03:04,008] - TelnetServer server is running on port 31000
  INFO root [2019-09-14 16:03:04,608] - ---- loginapp is running ----
 DEBUG root [2019-09-14 16:03:04,612] - Components::process(): Request for the process of identity...
  WARN root [2019-09-14 16:03:10,778] - Forwarding logs to logger[192.168.56.1:51709/0/0/0]...
 ERROR root [2019-09-14 16:06:45,794] - Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=37917357501, lastReceivedTime=37888477600, inactivityExceptionPeriod=134887437)
  INFO root [2019-09-14 16:06:45,804] - The following logs sent to logger failed:
  WARN root [2019-09-14 16:06:45,805] - ==> [2019-09-14 16:06:45 793] TCPPacketReceiver::processPendingEvents(192.168.56.1:51709): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-14 16:06:45,805] - ==> [2019-09-14 16:06:45 796] Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=37917357501, lastReceivedTime=37888477600, inactivityExceptionPeriod=134887437)
  WARN root [2019-09-14 16:06:45,904] - TCPPacketReceiver::processPendingEvents(192.168.56.1:51712): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-14 16:06:45,909] - Components::removeComponentByChannel: dbmgr : 4000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=37917614146, lastReceivedTime=37888576568, inactivityExceptionPeriod=134887437)
  INFO root [2019-09-14 16:06:45,914] - Shutdowner::onShutdownBegin: shutting down(period=0, tickPeriod=1)
S_INFO root [2019-09-14 16:06:45,923] - onLoginAppShutDown()
  INFO root [2019-09-14 16:06:45,923] - Shutdowner::onShutdown: shutting down
  INFO root [2019-09-14 16:06:45,923] - Shutdowner::onShutdownEnd: shutting down
  INFO root [2019-09-14 16:06:45,936] - Script::uninstall(): is successfully!
 DEBUG root [2019-09-14 16:06:45,936] - ThreadPool::destroy(): starting size 2.
  WARN root [2019-09-14 16:06:46,237] - ThreadPool::destroy(): waiting for thread(2)[0x1fe3a78,0x1fe40f8,], try=1
  INFO root [2019-09-14 16:06:47,718] - 
  INFO root [2019-09-14 16:06:47,729] - -----------------------------------------------------------------------------------------


  INFO root [2019-09-14 16:06:47,735] - Resmgr::initialize: KBE_ROOT=D:/kbengine/
  INFO root [2019-09-14 16:06:47,735] - Resmgr::initialize: KBE_RES_PATH=D:/kbengine/kbe/res/;D:/kbengine/BattleOfBalls_assets/;D:/kbengine/BattleOfBalls_assets/scripts/;D:/kbengine/BattleOfBalls_assets/res/
  INFO root [2019-09-14 16:06:47,735] - Resmgr::initialize: KBE_BIN_PATH=D:/kbengine/kbe/bin/server/
  INFO root [2019-09-14 16:06:47,768] - NetworkInterface::initialize(EXTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:20013 )
  INFO root [2019-09-14 16:06:47,768] - NetworkInterface::initialize(EXTERNAL-TCP): address 192.168.56.1:20013, SOMAXCONN=128.
  INFO root [2019-09-14 16:06:47,776] - NetworkInterface::initialize(INTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:51748 )
  INFO root [2019-09-14 16:06:47,776] - NetworkInterface::initialize(INTERNAL-TCP): address 192.168.56.1:51748, SOMAXCONN=128.
  INFO root [2019-09-14 16:06:47,776] - server-configs:
  INFO root [2019-09-14 16:06:47,776] - 	internalTcpAddr : 192.168.56.1:51748
  INFO root [2019-09-14 16:06:47,776] - 	externalTcpAddr : 192.168.56.1:20013
  INFO root [2019-09-14 16:06:47,777] - 	componentID : 9000
  WARN root [2019-09-14 16:06:48,016] - invalid UID(0) <= 0, please check UID for environment! automatically set to 2082.
  INFO root [2019-09-14 16:06:49,521] - ---- loginapp Version: 2.4.5. ScriptVersion: 0.1.0. Pythoncore: 3.7.3. Protocol: CE9FF53F4E85387EE4455C7191269324. Config: Debug X86. Built: 09:58:16 May  8 2019. AppID: 9000. UID: 2082. PID: 14152 ----
  INFO root [2019-09-14 16:06:49,521] - Server error_name: Intel(R) Core(TM) i5-6300HQ CPU @ 2.30GHz with 2097151kB RAM
  INFO root [2019-09-14 16:06:49,523] - ThreadPool::createThreadPool: creating  threadpool...
  INFO root [2019-09-14 16:06:49,523] - ThreadPool::createThreadPool: successfully(2), newThreadCount=1, normalMaxThreadCount=2, maxThreadCount=8
  INFO root [2019-09-14 16:06:49,785] - Script::install(): is successfully, Python=(3.7.3 (default, May  8 2019, 09:53:38) [MSC v.1916 32 bit (Intel)])!
  INFO root [2019-09-14 16:06:49,786] - Script::install(): KBExtra is successfully!
S_INFO root [2019-09-14 16:06:49,801] - onLoginAppReady: bootstrapGroupIndex=0, bootstrapGlobalIndex=0
  INFO root [2019-09-14 16:06:49,802] - TelnetServer server is running on port 31000
  INFO root [2019-09-14 16:06:50,252] - ---- loginapp is running ----
 DEBUG root [2019-09-14 16:06:50,283] - Components::process(): Request for the process of identity...
  INFO root [2019-09-14 16:06:57,442] - Components::findComponents: find logger(1)...
  INFO root [2019-09-14 16:06:57,456] - Components::findComponents: found logger, addr:192.168.56.1:51744
  INFO root [2019-09-14 16:06:57,456] - Components::addComponent: reset orderLog, uid=2082!
  INFO root [2019-09-14 16:06:57,456] - Components::addComponent[logger], uid=2082, componentID=2000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2019-09-14 16:06:57,458] - Components::findComponents: find dbmgr(2)...
  INFO root [2019-09-14 16:06:57,561] - Components::findComponents: found dbmgr, addr:192.168.56.1:51746
  INFO root [2019-09-14 16:06:57,561] - Components::addComponent: reset orderLog, uid=2082!
  INFO root [2019-09-14 16:06:57,561] - Components::addComponent[dbmgr], uid=2082, componentID=4000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2019-09-14 16:06:57,561] - Components::findComponents: found dbmgr, addr:192.168.56.1:51746
  WARN root [2019-09-14 16:06:58,562] - Forwarding logs to logger[192.168.56.1:51744/0/0/0]...
 ERROR root [2019-09-14 16:10:16,202] - Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=38390775229, lastReceivedTime=38326854982, inactivityExceptionPeriod=134887564)
  INFO root [2019-09-14 16:10:16,207] - The following logs sent to logger failed:
  WARN root [2019-09-14 16:10:16,208] - ==> [2019-09-14 16:10:16 200] TCPPacketReceiver::processPendingEvents(192.168.56.1:51744): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-14 16:10:16,208] - ==> [2019-09-14 16:10:16 204] Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=38390775229, lastReceivedTime=38326854982, inactivityExceptionPeriod=134887564)
  WARN root [2019-09-14 16:10:16,331] - TCPPacketReceiver::processPendingEvents(192.168.56.1:51746): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-14 16:10:16,342] - Components::removeComponentByChannel: dbmgr : 4000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=38391088274, lastReceivedTime=38326732042, inactivityExceptionPeriod=134887564)
  INFO root [2019-09-14 16:10:16,376] - Shutdowner::onShutdownBegin: shutting down(period=0, tickPeriod=1)
S_INFO root [2019-09-14 16:10:16,377] - onLoginAppShutDown()
  INFO root [2019-09-14 16:10:16,377] - Shutdowner::onShutdown: shutting down
  INFO root [2019-09-14 16:10:16,378] - Shutdowner::onShutdownEnd: shutting down
  INFO root [2019-09-14 16:10:16,392] - Script::uninstall(): is successfully!
 DEBUG root [2019-09-14 16:10:16,392] - ThreadPool::destroy(): starting size 2.
  WARN root [2019-09-14 16:10:16,693] - ThreadPool::destroy(): waiting for thread(2)[0x1f1dcf8,0x1f1e038,], try=1
  INFO root [2019-09-14 16:46:30,227] - 
  INFO root [2019-09-14 16:46:30,231] - -----------------------------------------------------------------------------------------


  INFO root [2019-09-14 16:46:30,232] - Resmgr::initialize: KBE_ROOT=D:/kbengine/
  INFO root [2019-09-14 16:46:30,233] - Resmgr::initialize: KBE_RES_PATH=D:/kbengine/kbe/res/;D:/kbengine/BattleOfBalls_assets/;D:/kbengine/BattleOfBalls_assets/scripts/;D:/kbengine/BattleOfBalls_assets/res/
  INFO root [2019-09-14 16:46:30,233] - Resmgr::initialize: KBE_BIN_PATH=D:/kbengine/kbe/bin/server/
  INFO root [2019-09-14 16:46:30,292] - NetworkInterface::initialize(EXTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:20013 )
  INFO root [2019-09-14 16:46:30,292] - NetworkInterface::initialize(EXTERNAL-TCP): address 192.168.56.1:20013, SOMAXCONN=128.
  INFO root [2019-09-14 16:46:30,297] - NetworkInterface::initialize(INTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:51801 )
  INFO root [2019-09-14 16:46:30,297] - NetworkInterface::initialize(INTERNAL-TCP): address 192.168.56.1:51801, SOMAXCONN=128.
  INFO root [2019-09-14 16:46:30,297] - server-configs:
  INFO root [2019-09-14 16:46:30,297] - 	internalTcpAddr : 192.168.56.1:51801
  INFO root [2019-09-14 16:46:30,297] - 	externalTcpAddr : 192.168.56.1:20013
  INFO root [2019-09-14 16:46:30,297] - 	componentID : 9000
  WARN root [2019-09-14 16:46:30,496] - invalid UID(0) <= 0, please check UID for environment! automatically set to 2082.
  INFO root [2019-09-14 16:46:32,001] - ---- loginapp Version: 2.4.5. ScriptVersion: 0.1.0. Pythoncore: 3.7.3. Protocol: CE9FF53F4E85387EE4455C7191269324. Config: Debug X86. Built: 09:58:16 May  8 2019. AppID: 9000. UID: 2082. PID: 14548 ----
  INFO root [2019-09-14 16:46:32,008] - Server error_name: Intel(R) Core(TM) i5-6300HQ CPU @ 2.30GHz with 2097151kB RAM
  INFO root [2019-09-14 16:46:32,036] - ThreadPool::createThreadPool: creating  threadpool...
  INFO root [2019-09-14 16:46:32,037] - ThreadPool::createThreadPool: successfully(2), newThreadCount=1, normalMaxThreadCount=2, maxThreadCount=8
  INFO root [2019-09-14 16:46:32,330] - Script::install(): is successfully, Python=(3.7.3 (default, May  8 2019, 09:53:38) [MSC v.1916 32 bit (Intel)])!
  INFO root [2019-09-14 16:46:32,330] - Script::install(): KBExtra is successfully!
S_INFO root [2019-09-14 16:46:32,342] - onLoginAppReady: bootstrapGroupIndex=0, bootstrapGlobalIndex=0
  INFO root [2019-09-14 16:46:32,343] - TelnetServer server is running on port 31000
  INFO root [2019-09-14 16:46:32,758] - ---- loginapp is running ----
 DEBUG root [2019-09-14 16:46:32,760] - Components::process(): Request for the process of identity...
  INFO root [2019-09-14 16:46:39,903] - Components::findComponents: find logger(1)...
  INFO root [2019-09-14 16:46:40,001] - Components::findComponents: found logger, addr:192.168.56.1:51797
  INFO root [2019-09-14 16:46:40,001] - Components::addComponent: reset orderLog, uid=2082!
  INFO root [2019-09-14 16:46:40,001] - Components::addComponent[logger], uid=2082, componentID=2000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2019-09-14 16:46:40,003] - Components::findComponents: find dbmgr(2)...
  INFO root [2019-09-14 16:46:40,009] - Components::findComponents: found dbmgr, addr:192.168.56.1:51799
  INFO root [2019-09-14 16:46:40,009] - Components::addComponent: reset orderLog, uid=2082!
  INFO root [2019-09-14 16:46:40,009] - Components::addComponent[dbmgr], uid=2082, componentID=4000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2019-09-14 16:46:40,112] - Components::findComponents: found dbmgr, addr:192.168.56.1:51799
  WARN root [2019-09-14 16:46:41,113] - Forwarding logs to logger[192.168.56.1:51797/0/0/0]...
 ERROR root [2019-09-14 16:47:58,732] - Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=43481468618, lastReceivedTime=43417384635, inactivityExceptionPeriod=134887628)
  INFO root [2019-09-14 16:47:58,737] - The following logs sent to logger failed:
  WARN root [2019-09-14 16:47:58,738] - ==> [2019-09-14 16:47:58 730] TCPPacketReceiver::processPendingEvents(192.168.56.1:51797): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-14 16:47:58,738] - ==> [2019-09-14 16:47:58 733] Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=43481468618, lastReceivedTime=43417384635, inactivityExceptionPeriod=134887628)
  WARN root [2019-09-14 16:47:58,846] - TCPPacketReceiver::processPendingEvents(192.168.56.1:51799): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-14 16:47:58,850] - Components::removeComponentByChannel: dbmgr : 4000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=43481733557, lastReceivedTime=43417446921, inactivityExceptionPeriod=134887628)
  INFO root [2019-09-14 16:47:58,863] - Shutdowner::onShutdownBegin: shutting down(period=0, tickPeriod=1)
S_INFO root [2019-09-14 16:47:58,878] - onLoginAppShutDown()
  INFO root [2019-09-14 16:47:58,879] - Shutdowner::onShutdown: shutting down
  INFO root [2019-09-14 16:47:58,879] - Shutdowner::onShutdownEnd: shutting down
  INFO root [2019-09-14 16:47:58,891] - Script::uninstall(): is successfully!
 DEBUG root [2019-09-14 16:47:58,891] - ThreadPool::destroy(): starting size 2.
  WARN root [2019-09-14 16:47:59,191] - ThreadPool::destroy(): waiting for thread(2)[0x1ff34e8,0x1ff3828,], try=1
  INFO root [2019-09-14 16:48:00,585] - 
  INFO root [2019-09-14 16:48:00,590] - -----------------------------------------------------------------------------------------


  INFO root [2019-09-14 16:48:00,591] - Resmgr::initialize: KBE_ROOT=D:/kbengine/
  INFO root [2019-09-14 16:48:00,592] - Resmgr::initialize: KBE_RES_PATH=D:/kbengine/kbe/res/;D:/kbengine/BattleOfBalls_assets/;D:/kbengine/BattleOfBalls_assets/scripts/;D:/kbengine/BattleOfBalls_assets/res/
  INFO root [2019-09-14 16:48:00,592] - Resmgr::initialize: KBE_BIN_PATH=D:/kbengine/kbe/bin/server/
  INFO root [2019-09-14 16:48:00,653] - NetworkInterface::initialize(EXTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:20013 )
  INFO root [2019-09-14 16:48:00,653] - NetworkInterface::initialize(EXTERNAL-TCP): address 192.168.56.1:20013, SOMAXCONN=128.
  INFO root [2019-09-14 16:48:00,664] - NetworkInterface::initialize(INTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:51837 )
  INFO root [2019-09-14 16:48:00,664] - NetworkInterface::initialize(INTERNAL-TCP): address 192.168.56.1:51837, SOMAXCONN=128.
  INFO root [2019-09-14 16:48:00,664] - server-configs:
  INFO root [2019-09-14 16:48:00,665] - 	internalTcpAddr : 192.168.56.1:51837
  INFO root [2019-09-14 16:48:00,665] - 	externalTcpAddr : 192.168.56.1:20013
  INFO root [2019-09-14 16:48:00,665] - 	componentID : 9000
  WARN root [2019-09-14 16:48:00,966] - invalid UID(0) <= 0, please check UID for environment! automatically set to 2082.
  INFO root [2019-09-14 16:48:01,370] - Components::findLogger: found logger, addr:192.168.56.1:51831
  INFO root [2019-09-14 16:48:01,370] - Components::addComponent: reset orderLog, uid=2082!
  INFO root [2019-09-14 16:48:01,371] - Components::addComponent[logger], uid=2082, componentID=2000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2019-09-14 16:48:01,376] - ---- loginapp Version: 2.4.5. ScriptVersion: 0.1.0. Pythoncore: 3.7.3. Protocol: CE9FF53F4E85387EE4455C7191269324. Config: Debug X86. Built: 09:58:16 May  8 2019. AppID: 9000. UID: 2082. PID: 7028 ----
  INFO root [2019-09-14 16:48:01,377] - Server error_name: Intel(R) Core(TM) i5-6300HQ CPU @ 2.30GHz with 2097151kB RAM
  INFO root [2019-09-14 16:48:01,378] - ThreadPool::createThreadPool: creating  threadpool...
  INFO root [2019-09-14 16:48:01,378] - ThreadPool::createThreadPool: successfully(2), newThreadCount=1, normalMaxThreadCount=2, maxThreadCount=8
  INFO root [2019-09-14 16:48:01,601] - Script::install(): is successfully, Python=(3.7.3 (default, May  8 2019, 09:53:38) [MSC v.1916 32 bit (Intel)])!
  INFO root [2019-09-14 16:48:01,601] - Script::install(): KBExtra is successfully!
S_INFO root [2019-09-14 16:48:01,612] - onLoginAppReady: bootstrapGroupIndex=0, bootstrapGlobalIndex=0
  INFO root [2019-09-14 16:48:01,613] - TelnetServer server is running on port 31000
  INFO root [2019-09-14 16:48:02,061] - ---- loginapp is running ----
 DEBUG root [2019-09-14 16:48:02,072] - Components::process(): Request for the process of identity...
  WARN root [2019-09-14 16:48:08,293] - Forwarding logs to logger[192.168.56.1:51831/0/0/0]...
 ERROR root [2019-09-14 16:52:19,730] - Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=44068713834, lastReceivedTime=44025401727, inactivityExceptionPeriod=134887564)
  INFO root [2019-09-14 16:52:19,732] - The following logs sent to logger failed:
  WARN root [2019-09-14 16:52:19,732] - ==> [2019-09-14 16:52:19 726] TCPPacketReceiver::processPendingEvents(192.168.56.1:51831): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-14 16:52:19,733] - ==> [2019-09-14 16:52:19 730] Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=44068713834, lastReceivedTime=44025401727, inactivityExceptionPeriod=134887564)
  WARN root [2019-09-14 16:52:19,823] - TCPPacketReceiver::processPendingEvents(192.168.56.1:51832): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-14 16:52:19,830] - Components::removeComponentByChannel: dbmgr : 4000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=44068940676, lastReceivedTime=44025508990, inactivityExceptionPeriod=134887564)
  INFO root [2019-09-14 16:52:19,834] - Shutdowner::onShutdownBegin: shutting down(period=0, tickPeriod=1)
S_INFO root [2019-09-14 16:52:19,836] - onLoginAppShutDown()
  INFO root [2019-09-14 16:52:19,837] - Shutdowner::onShutdown: shutting down
  INFO root [2019-09-14 16:52:19,837] - Shutdowner::onShutdownEnd: shutting down
  INFO root [2019-09-14 16:52:19,845] - Script::uninstall(): is successfully!
 DEBUG root [2019-09-14 16:52:19,845] - ThreadPool::destroy(): starting size 2.
  WARN root [2019-09-14 16:52:20,146] - ThreadPool::destroy(): waiting for thread(2)[0x1f559f0,0x1f56480,], try=1
  INFO root [2019-09-14 16:52:21,605] - 
  INFO root [2019-09-14 16:52:21,613] - -----------------------------------------------------------------------------------------


  INFO root [2019-09-14 16:52:21,614] - Resmgr::initialize: KBE_ROOT=D:/kbengine/
  INFO root [2019-09-14 16:52:21,614] - Resmgr::initialize: KBE_RES_PATH=D:/kbengine/kbe/res/;D:/kbengine/BattleOfBalls_assets/;D:/kbengine/BattleOfBalls_assets/scripts/;D:/kbengine/BattleOfBalls_assets/res/
  INFO root [2019-09-14 16:52:21,614] - Resmgr::initialize: KBE_BIN_PATH=D:/kbengine/kbe/bin/server/
  INFO root [2019-09-14 16:52:21,651] - NetworkInterface::initialize(EXTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:20013 )
  INFO root [2019-09-14 16:52:21,651] - NetworkInterface::initialize(EXTERNAL-TCP): address 192.168.56.1:20013, SOMAXCONN=128.
  INFO root [2019-09-14 16:52:21,660] - NetworkInterface::initialize(INTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:51876 )
  INFO root [2019-09-14 16:52:21,661] - NetworkInterface::initialize(INTERNAL-TCP): address 192.168.56.1:51876, SOMAXCONN=128.
  INFO root [2019-09-14 16:52:21,661] - server-configs:
  INFO root [2019-09-14 16:52:21,661] - 	internalTcpAddr : 192.168.56.1:51876
  INFO root [2019-09-14 16:52:21,661] - 	externalTcpAddr : 192.168.56.1:20013
  INFO root [2019-09-14 16:52:21,661] - 	componentID : 9000
  WARN root [2019-09-14 16:52:21,888] - invalid UID(0) <= 0, please check UID for environment! automatically set to 2082.
  INFO root [2019-09-14 16:52:22,158] - Components::findLogger: found logger, addr:192.168.56.1:51871
  INFO root [2019-09-14 16:52:22,158] - Components::addComponent: reset orderLog, uid=2082!
  INFO root [2019-09-14 16:52:22,158] - Components::addComponent[logger], uid=2082, componentID=2000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2019-09-14 16:52:22,164] - ---- loginapp Version: 2.4.5. ScriptVersion: 0.1.0. Pythoncore: 3.7.3. Protocol: CE9FF53F4E85387EE4455C7191269324. Config: Debug X86. Built: 09:58:16 May  8 2019. AppID: 9000. UID: 2082. PID: 15264 ----
  INFO root [2019-09-14 16:52:22,164] - Server error_name: Intel(R) Core(TM) i5-6300HQ CPU @ 2.30GHz with 2097151kB RAM
  INFO root [2019-09-14 16:52:22,164] - ThreadPool::createThreadPool: creating  threadpool...
  INFO root [2019-09-14 16:52:22,164] - ThreadPool::createThreadPool: successfully(2), newThreadCount=1, normalMaxThreadCount=2, maxThreadCount=8
  INFO root [2019-09-14 16:52:22,362] - Script::install(): is successfully, Python=(3.7.3 (default, May  8 2019, 09:53:38) [MSC v.1916 32 bit (Intel)])!
  INFO root [2019-09-14 16:52:22,362] - Script::install(): KBExtra is successfully!
S_INFO root [2019-09-14 16:52:22,374] - onLoginAppReady: bootstrapGroupIndex=0, bootstrapGlobalIndex=0
  INFO root [2019-09-14 16:52:22,375] - TelnetServer server is running on port 31000
  INFO root [2019-09-14 16:52:22,800] - ---- loginapp is running ----
 DEBUG root [2019-09-14 16:52:22,801] - Components::process(): Request for the process of identity...
  WARN root [2019-09-14 16:52:28,929] - Forwarding logs to logger[192.168.56.1:51871/0/0/0]...
 ERROR root [2019-09-14 16:55:22,853] - Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=44480742009, lastReceivedTime=44477724575, inactivityExceptionPeriod=134887564)
  INFO root [2019-09-14 16:55:22,859] - The following logs sent to logger failed:
  WARN root [2019-09-14 16:55:22,860] - ==> [2019-09-14 16:55:22 850] TCPPacketReceiver::processPendingEvents(192.168.56.1:51871): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-14 16:55:22,860] - ==> [2019-09-14 16:55:22 854] Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=44480742009, lastReceivedTime=44477724575, inactivityExceptionPeriod=134887564)
  WARN root [2019-09-14 16:55:22,968] - TCPPacketReceiver::processPendingEvents(192.168.56.1:51869): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-14 16:55:22,972] - Components::removeComponentByChannel: dbmgr : 4000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=44481007288, lastReceivedTime=44477686078, inactivityExceptionPeriod=134887564)
  INFO root [2019-09-14 16:55:22,995] - Shutdowner::onShutdownBegin: shutting down(period=0, tickPeriod=1)
S_INFO root [2019-09-14 16:55:23,000] - onLoginAppShutDown()
  INFO root [2019-09-14 16:55:23,001] - Shutdowner::onShutdown: shutting down
  INFO root [2019-09-14 16:55:23,001] - Shutdowner::onShutdownEnd: shutting down
  INFO root [2019-09-14 16:55:23,018] - Script::uninstall(): is successfully!
 DEBUG root [2019-09-14 16:55:23,018] - ThreadPool::destroy(): starting size 2.
  WARN root [2019-09-14 16:55:23,320] - ThreadPool::destroy(): waiting for thread(2)[0x73a580,0x73a518,], try=1
  INFO root [2019-09-14 16:57:03,864] - 
  INFO root [2019-09-14 16:57:03,902] - -----------------------------------------------------------------------------------------


  INFO root [2019-09-14 16:57:03,903] - Resmgr::initialize: KBE_ROOT=D:/kbengine/
  INFO root [2019-09-14 16:57:03,903] - Resmgr::initialize: KBE_RES_PATH=D:/kbengine/kbe/res/;D:/kbengine/BattleOfBalls_assets/;D:/kbengine/BattleOfBalls_assets/scripts/;D:/kbengine/BattleOfBalls_assets/res/
  INFO root [2019-09-14 16:57:03,904] - Resmgr::initialize: KBE_BIN_PATH=D:/kbengine/kbe/bin/server/
  INFO root [2019-09-14 16:57:03,963] - NetworkInterface::initialize(EXTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:20013 )
  INFO root [2019-09-14 16:57:03,964] - NetworkInterface::initialize(EXTERNAL-TCP): address 192.168.56.1:20013, SOMAXCONN=128.
  INFO root [2019-09-14 16:57:03,973] - NetworkInterface::initialize(INTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:51915 )
  INFO root [2019-09-14 16:57:03,974] - NetworkInterface::initialize(INTERNAL-TCP): address 192.168.56.1:51915, SOMAXCONN=128.
  INFO root [2019-09-14 16:57:03,976] - server-configs:
  INFO root [2019-09-14 16:57:03,976] - 	internalTcpAddr : 192.168.56.1:51915
  INFO root [2019-09-14 16:57:03,976] - 	externalTcpAddr : 192.168.56.1:20013
  INFO root [2019-09-14 16:57:03,976] - 	componentID : 9000
  WARN root [2019-09-14 16:57:04,296] - invalid UID(0) <= 0, please check UID for environment! automatically set to 2082.
  INFO root [2019-09-14 16:57:04,610] - Components::findLogger: found logger, addr:192.168.56.1:51912
  INFO root [2019-09-14 16:57:04,610] - Components::addComponent: reset orderLog, uid=2082!
  INFO root [2019-09-14 16:57:04,610] - Components::addComponent[logger], uid=2082, componentID=2000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2019-09-14 16:57:04,616] - ---- loginapp Version: 2.4.5. ScriptVersion: 0.1.0. Pythoncore: 3.7.3. Protocol: CE9FF53F4E85387EE4455C7191269324. Config: Debug X86. Built: 09:58:16 May  8 2019. AppID: 9000. UID: 2082. PID: 14260 ----
  INFO root [2019-09-14 16:57:04,616] - Server error_name: Intel(R) Core(TM) i5-6300HQ CPU @ 2.30GHz with 2097151kB RAM
  INFO root [2019-09-14 16:57:04,616] - ThreadPool::createThreadPool: creating  threadpool...
  INFO root [2019-09-14 16:57:04,617] - ThreadPool::createThreadPool: successfully(2), newThreadCount=1, normalMaxThreadCount=2, maxThreadCount=8
  INFO root [2019-09-14 16:57:04,891] - Script::install(): is successfully, Python=(3.7.3 (default, May  8 2019, 09:53:38) [MSC v.1916 32 bit (Intel)])!
  INFO root [2019-09-14 16:57:04,891] - Script::install(): KBExtra is successfully!
S_INFO root [2019-09-14 16:57:04,907] - onLoginAppReady: bootstrapGroupIndex=0, bootstrapGlobalIndex=0
  INFO root [2019-09-14 16:57:04,907] - TelnetServer server is running on port 31000
  INFO root [2019-09-14 16:57:05,391] - ---- loginapp is running ----
 DEBUG root [2019-09-14 16:57:05,392] - Components::process(): Request for the process of identity...
  WARN root [2019-09-14 16:57:11,539] - Forwarding logs to logger[192.168.56.1:51912/0/0/0]...
 ERROR root [2019-09-14 16:58:29,267] - Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=44900174454, lastReceivedTime=44842788745, inactivityExceptionPeriod=134887437)
  INFO root [2019-09-14 16:58:29,272] - The following logs sent to logger failed:
  WARN root [2019-09-14 16:58:29,273] - ==> [2019-09-14 16:58:29 266] TCPPacketReceiver::processPendingEvents(192.168.56.1:51912): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-14 16:58:29,273] - ==> [2019-09-14 16:58:29 268] Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=44900174454, lastReceivedTime=44842788745, inactivityExceptionPeriod=134887437)
  WARN root [2019-09-14 16:58:29,369] - TCPPacketReceiver::processPendingEvents(192.168.56.1:51914): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-14 16:58:29,371] - Components::removeComponentByChannel: dbmgr : 4000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=44900407962, lastReceivedTime=44843134038, inactivityExceptionPeriod=134887437)
  INFO root [2019-09-14 16:58:29,375] - Shutdowner::onShutdownBegin: shutting down(period=0, tickPeriod=1)
S_INFO root [2019-09-14 16:58:29,376] - onLoginAppShutDown()
  INFO root [2019-09-14 16:58:29,376] - Shutdowner::onShutdown: shutting down
  INFO root [2019-09-14 16:58:29,377] - Shutdowner::onShutdownEnd: shutting down
  INFO root [2019-09-14 16:58:29,385] - Script::uninstall(): is successfully!
 DEBUG root [2019-09-14 16:58:29,385] - ThreadPool::destroy(): starting size 2.
  WARN root [2019-09-14 16:58:29,686] - ThreadPool::destroy(): waiting for thread(2)[0x9445d0,0x943938,], try=1
  INFO root [2019-09-14 17:01:33,314] - 
  INFO root [2019-09-14 17:01:33,324] - -----------------------------------------------------------------------------------------


  INFO root [2019-09-14 17:01:33,324] - Resmgr::initialize: KBE_ROOT=D:/kbengine/
  INFO root [2019-09-14 17:01:33,325] - Resmgr::initialize: KBE_RES_PATH=D:/kbengine/kbe/res/;D:/kbengine/BattleOfBalls_assets/;D:/kbengine/BattleOfBalls_assets/scripts/;D:/kbengine/BattleOfBalls_assets/res/
  INFO root [2019-09-14 17:01:33,325] - Resmgr::initialize: KBE_BIN_PATH=D:/kbengine/kbe/bin/server/
  INFO root [2019-09-14 17:01:33,361] - NetworkInterface::initialize(EXTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:20013 )
  INFO root [2019-09-14 17:01:33,361] - NetworkInterface::initialize(EXTERNAL-TCP): address 192.168.56.1:20013, SOMAXCONN=128.
  INFO root [2019-09-14 17:01:33,366] - NetworkInterface::initialize(INTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:51953 )
  INFO root [2019-09-14 17:01:33,366] - NetworkInterface::initialize(INTERNAL-TCP): address 192.168.56.1:51953, SOMAXCONN=128.
  INFO root [2019-09-14 17:01:33,366] - server-configs:
  INFO root [2019-09-14 17:01:33,367] - 	internalTcpAddr : 192.168.56.1:51953
  INFO root [2019-09-14 17:01:33,367] - 	externalTcpAddr : 192.168.56.1:20013
  INFO root [2019-09-14 17:01:33,367] - 	componentID : 9000
  WARN root [2019-09-14 17:01:33,574] - invalid UID(0) <= 0, please check UID for environment! automatically set to 2082.
  INFO root [2019-09-14 17:01:33,906] - Components::findLogger: found logger, addr:192.168.56.1:51949
  INFO root [2019-09-14 17:01:33,907] - Components::addComponent: reset orderLog, uid=2082!
  INFO root [2019-09-14 17:01:33,907] - Components::addComponent[logger], uid=2082, componentID=2000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2019-09-14 17:01:33,912] - ---- loginapp Version: 2.4.5. ScriptVersion: 0.1.0. Pythoncore: 3.7.3. Protocol: CE9FF53F4E85387EE4455C7191269324. Config: Debug X86. Built: 09:58:16 May  8 2019. AppID: 9000. UID: 2082. PID: 3888 ----
  INFO root [2019-09-14 17:01:33,913] - Server error_name: Intel(R) Core(TM) i5-6300HQ CPU @ 2.30GHz with 2097151kB RAM
  INFO root [2019-09-14 17:01:33,914] - ThreadPool::createThreadPool: creating  threadpool...
  INFO root [2019-09-14 17:01:33,914] - ThreadPool::createThreadPool: successfully(2), newThreadCount=1, normalMaxThreadCount=2, maxThreadCount=8
  INFO root [2019-09-14 17:01:34,117] - Script::install(): is successfully, Python=(3.7.3 (default, May  8 2019, 09:53:38) [MSC v.1916 32 bit (Intel)])!
  INFO root [2019-09-14 17:01:34,118] - Script::install(): KBExtra is successfully!
S_INFO root [2019-09-14 17:01:34,129] - onLoginAppReady: bootstrapGroupIndex=0, bootstrapGlobalIndex=0
  INFO root [2019-09-14 17:01:34,129] - TelnetServer server is running on port 31000
  INFO root [2019-09-14 17:01:34,531] - ---- loginapp is running ----
 DEBUG root [2019-09-14 17:01:34,532] - Components::process(): Request for the process of identity...
  WARN root [2019-09-14 17:01:40,667] - Forwarding logs to logger[192.168.56.1:51949/0/0/0]...
 ERROR root [2019-09-14 17:03:29,806] - Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=45576385289, lastReceivedTime=45516723262, inactivityExceptionPeriod=134887197)
  INFO root [2019-09-14 17:03:29,813] - The following logs sent to logger failed:
  WARN root [2019-09-14 17:03:29,814] - ==> [2019-09-14 17:03:29 802] TCPPacketReceiver::processPendingEvents(192.168.56.1:51949): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-14 17:03:29,814] - ==> [2019-09-14 17:03:29 807] Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=45576385289, lastReceivedTime=45516723262, inactivityExceptionPeriod=134887197)
  WARN root [2019-09-14 17:03:29,914] - TCPPacketReceiver::processPendingEvents(192.168.56.1:51955): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-14 17:03:29,923] - Components::removeComponentByChannel: dbmgr : 4000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=45576649220, lastReceivedTime=45516954159, inactivityExceptionPeriod=134887197)
  INFO root [2019-09-14 17:03:29,933] - Shutdowner::onShutdownBegin: shutting down(period=0, tickPeriod=1)
S_INFO root [2019-09-14 17:03:29,935] - onLoginAppShutDown()
  INFO root [2019-09-14 17:03:29,936] - Shutdowner::onShutdown: shutting down
  INFO root [2019-09-14 17:03:29,936] - Shutdowner::onShutdownEnd: shutting down
  INFO root [2019-09-14 17:03:29,949] - Script::uninstall(): is successfully!
 DEBUG root [2019-09-14 17:03:29,950] - ThreadPool::destroy(): starting size 2.
  WARN root [2019-09-14 17:03:30,251] - ThreadPool::destroy(): waiting for thread(2)[0x1b3e50,0x1b4468,], try=1
  INFO root [2019-09-14 17:03:31,764] - 
  INFO root [2019-09-14 17:03:31,764] - -----------------------------------------------------------------------------------------


  INFO root [2019-09-14 17:03:31,765] - Resmgr::initialize: KBE_ROOT=D:/kbengine/
  INFO root [2019-09-14 17:03:31,766] - Resmgr::initialize: KBE_RES_PATH=D:/kbengine/kbe/res/;D:/kbengine/BattleOfBalls_assets/;D:/kbengine/BattleOfBalls_assets/scripts/;D:/kbengine/BattleOfBalls_assets/res/
  INFO root [2019-09-14 17:03:31,766] - Resmgr::initialize: KBE_BIN_PATH=D:/kbengine/kbe/bin/server/
  INFO root [2019-09-14 17:03:31,784] - NetworkInterface::initialize(EXTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:20013 )
  INFO root [2019-09-14 17:03:31,784] - NetworkInterface::initialize(EXTERNAL-TCP): address 192.168.56.1:20013, SOMAXCONN=128.
  INFO root [2019-09-14 17:03:31,793] - NetworkInterface::initialize(INTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:51993 )
  INFO root [2019-09-14 17:03:31,793] - NetworkInterface::initialize(INTERNAL-TCP): address 192.168.56.1:51993, SOMAXCONN=128.
  INFO root [2019-09-14 17:03:31,793] - server-configs:
  INFO root [2019-09-14 17:03:31,793] - 	internalTcpAddr : 192.168.56.1:51993
  INFO root [2019-09-14 17:03:31,793] - 	externalTcpAddr : 192.168.56.1:20013
  INFO root [2019-09-14 17:03:31,793] - 	componentID : 9000
  WARN root [2019-09-14 17:03:32,010] - invalid UID(0) <= 0, please check UID for environment! automatically set to 2082.
  INFO root [2019-09-14 17:03:32,317] - Components::findLogger: found logger, addr:192.168.56.1:51994
  INFO root [2019-09-14 17:03:32,317] - Components::addComponent: reset orderLog, uid=2082!
  INFO root [2019-09-14 17:03:32,317] - Components::addComponent[logger], uid=2082, componentID=2000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2019-09-14 17:03:32,323] - ---- loginapp Version: 2.4.5. ScriptVersion: 0.1.0. Pythoncore: 3.7.3. Protocol: CE9FF53F4E85387EE4455C7191269324. Config: Debug X86. Built: 09:58:16 May  8 2019. AppID: 9000. UID: 2082. PID: 13692 ----
  INFO root [2019-09-14 17:03:32,323] - Server error_name: Intel(R) Core(TM) i5-6300HQ CPU @ 2.30GHz with 2097151kB RAM
  INFO root [2019-09-14 17:03:32,323] - ThreadPool::createThreadPool: creating  threadpool...
  INFO root [2019-09-14 17:03:32,324] - ThreadPool::createThreadPool: successfully(2), newThreadCount=1, normalMaxThreadCount=2, maxThreadCount=8
  INFO root [2019-09-14 17:03:32,523] - Script::install(): is successfully, Python=(3.7.3 (default, May  8 2019, 09:53:38) [MSC v.1916 32 bit (Intel)])!
  INFO root [2019-09-14 17:03:32,524] - Script::install(): KBExtra is successfully!
S_INFO root [2019-09-14 17:03:32,535] - onLoginAppReady: bootstrapGroupIndex=0, bootstrapGlobalIndex=0
  INFO root [2019-09-14 17:03:32,535] - TelnetServer server is running on port 31000
  INFO root [2019-09-14 17:03:32,915] - ---- loginapp is running ----
 DEBUG root [2019-09-14 17:03:32,915] - Components::process(): Request for the process of identity...
  WARN root [2019-09-14 17:03:39,027] - Forwarding logs to logger[192.168.56.1:51994/0/0/0]...
 ERROR root [2019-09-14 17:05:44,937] - Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=45880431737, lastReceivedTime=45850872417, inactivityExceptionPeriod=134887437)
  INFO root [2019-09-14 17:05:44,943] - The following logs sent to logger failed:
  WARN root [2019-09-14 17:05:44,943] - ==> [2019-09-14 17:05:44 934] TCPPacketReceiver::processPendingEvents(192.168.56.1:51994): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-14 17:05:44,943] - ==> [2019-09-14 17:05:44 938] Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=45880431737, lastReceivedTime=45850872417, inactivityExceptionPeriod=134887437)
  WARN root [2019-09-14 17:05:45,035] - TCPPacketReceiver::processPendingEvents(192.168.56.1:51987): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-14 17:05:45,040] - Components::removeComponentByChannel: dbmgr : 4000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=45880661434, lastReceivedTime=45850703594, inactivityExceptionPeriod=134887437)
  INFO root [2019-09-14 17:05:45,047] - Shutdowner::onShutdownBegin: shutting down(period=0, tickPeriod=1)
S_INFO root [2019-09-14 17:05:45,053] - onLoginAppShutDown()
  INFO root [2019-09-14 17:05:45,053] - Shutdowner::onShutdown: shutting down
  INFO root [2019-09-14 17:05:45,054] - Shutdowner::onShutdownEnd: shutting down
  INFO root [2019-09-14 17:05:45,062] - Script::uninstall(): is successfully!
 DEBUG root [2019-09-14 17:05:45,062] - ThreadPool::destroy(): starting size 2.
  WARN root [2019-09-14 17:05:45,363] - ThreadPool::destroy(): waiting for thread(2)[0x112ac0,0x112718,], try=1
  INFO root [2019-09-14 17:05:46,852] - 
  INFO root [2019-09-14 17:05:46,878] - -----------------------------------------------------------------------------------------


  INFO root [2019-09-14 17:05:46,879] - Resmgr::initialize: KBE_ROOT=D:/kbengine/
  INFO root [2019-09-14 17:05:46,879] - Resmgr::initialize: KBE_RES_PATH=D:/kbengine/kbe/res/;D:/kbengine/BattleOfBalls_assets/;D:/kbengine/BattleOfBalls_assets/scripts/;D:/kbengine/BattleOfBalls_assets/res/
  INFO root [2019-09-14 17:05:46,879] - Resmgr::initialize: KBE_BIN_PATH=D:/kbengine/kbe/bin/server/
  INFO root [2019-09-14 17:05:46,936] - NetworkInterface::initialize(EXTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:20013 )
  INFO root [2019-09-14 17:05:46,936] - NetworkInterface::initialize(EXTERNAL-TCP): address 192.168.56.1:20013, SOMAXCONN=128.
  INFO root [2019-09-14 17:05:46,943] - NetworkInterface::initialize(INTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:52033 )
  INFO root [2019-09-14 17:05:46,944] - NetworkInterface::initialize(INTERNAL-TCP): address 192.168.56.1:52033, SOMAXCONN=128.
  INFO root [2019-09-14 17:05:46,944] - server-configs:
  INFO root [2019-09-14 17:05:46,944] - 	internalTcpAddr : 192.168.56.1:52033
  INFO root [2019-09-14 17:05:46,944] - 	externalTcpAddr : 192.168.56.1:20013
  INFO root [2019-09-14 17:05:46,944] - 	componentID : 9000
  WARN root [2019-09-14 17:05:47,164] - invalid UID(0) <= 0, please check UID for environment! automatically set to 2082.
  INFO root [2019-09-14 17:05:47,648] - Components::findLogger: found logger, addr:192.168.56.1:52026
  INFO root [2019-09-14 17:05:47,648] - Components::addComponent: reset orderLog, uid=2082!
  INFO root [2019-09-14 17:05:47,649] - Components::addComponent[logger], uid=2082, componentID=2000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2019-09-14 17:05:47,654] - ---- loginapp Version: 2.4.5. ScriptVersion: 0.1.0. Pythoncore: 3.7.3. Protocol: CE9FF53F4E85387EE4455C7191269324. Config: Debug X86. Built: 09:58:16 May  8 2019. AppID: 9000. UID: 2082. PID: 13076 ----
  INFO root [2019-09-14 17:05:47,655] - Server error_name: Intel(R) Core(TM) i5-6300HQ CPU @ 2.30GHz with 2097151kB RAM
  INFO root [2019-09-14 17:05:47,655] - ThreadPool::createThreadPool: creating  threadpool...
  INFO root [2019-09-14 17:05:47,655] - ThreadPool::createThreadPool: successfully(2), newThreadCount=1, normalMaxThreadCount=2, maxThreadCount=8
  INFO root [2019-09-14 17:05:47,850] - Script::install(): is successfully, Python=(3.7.3 (default, May  8 2019, 09:53:38) [MSC v.1916 32 bit (Intel)])!
  INFO root [2019-09-14 17:05:47,851] - Script::install(): KBExtra is successfully!
S_INFO root [2019-09-14 17:05:47,862] - onLoginAppReady: bootstrapGroupIndex=0, bootstrapGlobalIndex=0
  INFO root [2019-09-14 17:05:47,862] - TelnetServer server is running on port 31000
  INFO root [2019-09-14 17:05:48,237] - ---- loginapp is running ----
 DEBUG root [2019-09-14 17:05:48,238] - Components::process(): Request for the process of identity...
  WARN root [2019-09-14 17:05:54,351] - Forwarding logs to logger[192.168.56.1:52026/0/0/0]...
 ERROR root [2019-09-14 17:09:12,047] - Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=46346427660, lastReceivedTime=46289473947, inactivityExceptionPeriod=134887564)
  INFO root [2019-09-14 17:09:12,050] - The following logs sent to logger failed:
  WARN root [2019-09-14 17:09:12,050] - ==> [2019-09-14 17:09:12 042] TCPPacketReceiver::processPendingEvents(192.168.56.1:52026): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-14 17:09:12,050] - ==> [2019-09-14 17:09:12 047] Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=46346427660, lastReceivedTime=46289473947, inactivityExceptionPeriod=134887564)
  WARN root [2019-09-14 17:09:12,155] - TCPPacketReceiver::processPendingEvents(192.168.56.1:52028): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-14 17:09:12,163] - Components::removeComponentByChannel: dbmgr : 4000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=46346689341, lastReceivedTime=46289864914, inactivityExceptionPeriod=134887564)
  INFO root [2019-09-14 17:09:12,182] - Shutdowner::onShutdownBegin: shutting down(period=0, tickPeriod=1)
S_INFO root [2019-09-14 17:09:12,184] - onLoginAppShutDown()
  INFO root [2019-09-14 17:09:12,184] - Shutdowner::onShutdown: shutting down
  INFO root [2019-09-14 17:09:12,184] - Shutdowner::onShutdownEnd: shutting down
  INFO root [2019-09-14 17:09:12,199] - Script::uninstall(): is successfully!
 DEBUG root [2019-09-14 17:09:12,199] - ThreadPool::destroy(): starting size 2.
  WARN root [2019-09-14 17:09:12,501] - ThreadPool::destroy(): waiting for thread(2)[0x1f1e1e0,0x1fa5de0,], try=1
  INFO root [2019-09-14 17:09:33,927] - 
  INFO root [2019-09-14 17:09:33,945] - -----------------------------------------------------------------------------------------


  INFO root [2019-09-14 17:09:33,946] - Resmgr::initialize: KBE_ROOT=D:/kbengine/
  INFO root [2019-09-14 17:09:33,946] - Resmgr::initialize: KBE_RES_PATH=D:/kbengine/kbe/res/;D:/kbengine/BattleOfBalls_assets/;D:/kbengine/BattleOfBalls_assets/scripts/;D:/kbengine/BattleOfBalls_assets/res/
  INFO root [2019-09-14 17:09:33,946] - Resmgr::initialize: KBE_BIN_PATH=D:/kbengine/kbe/bin/server/
  INFO root [2019-09-14 17:09:33,976] - NetworkInterface::initialize(EXTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:20013 )
  INFO root [2019-09-14 17:09:33,976] - NetworkInterface::initialize(EXTERNAL-TCP): address 192.168.56.1:20013, SOMAXCONN=128.
  INFO root [2019-09-14 17:09:33,980] - NetworkInterface::initialize(INTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:52070 )
  INFO root [2019-09-14 17:09:33,981] - NetworkInterface::initialize(INTERNAL-TCP): address 192.168.56.1:52070, SOMAXCONN=128.
  INFO root [2019-09-14 17:09:33,981] - server-configs:
  INFO root [2019-09-14 17:09:33,981] - 	internalTcpAddr : 192.168.56.1:52070
  INFO root [2019-09-14 17:09:33,981] - 	externalTcpAddr : 192.168.56.1:20013
  INFO root [2019-09-14 17:09:33,981] - 	componentID : 9000
  WARN root [2019-09-14 17:09:34,206] - invalid UID(0) <= 0, please check UID for environment! automatically set to 2082.
  INFO root [2019-09-14 17:09:34,413] - Components::findLogger: found logger, addr:192.168.56.1:52067
  INFO root [2019-09-14 17:09:34,413] - Components::addComponent: reset orderLog, uid=2082!
  INFO root [2019-09-14 17:09:34,414] - Components::addComponent[logger], uid=2082, componentID=2000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2019-09-14 17:09:34,423] - ---- loginapp Version: 2.4.5. ScriptVersion: 0.1.0. Pythoncore: 3.7.3. Protocol: CE9FF53F4E85387EE4455C7191269324. Config: Debug X86. Built: 09:58:16 May  8 2019. AppID: 9000. UID: 2082. PID: 15472 ----
  INFO root [2019-09-14 17:09:34,424] - Server error_name: Intel(R) Core(TM) i5-6300HQ CPU @ 2.30GHz with 2097151kB RAM
  INFO root [2019-09-14 17:09:34,424] - ThreadPool::createThreadPool: creating  threadpool...
  INFO root [2019-09-14 17:09:34,424] - ThreadPool::createThreadPool: successfully(2), newThreadCount=1, normalMaxThreadCount=2, maxThreadCount=8
  INFO root [2019-09-14 17:09:34,673] - Script::install(): is successfully, Python=(3.7.3 (default, May  8 2019, 09:53:38) [MSC v.1916 32 bit (Intel)])!
  INFO root [2019-09-14 17:09:34,674] - Script::install(): KBExtra is successfully!
S_INFO root [2019-09-14 17:09:34,685] - onLoginAppReady: bootstrapGroupIndex=0, bootstrapGlobalIndex=0
  INFO root [2019-09-14 17:09:34,685] - TelnetServer server is running on port 31000
  INFO root [2019-09-14 17:09:35,164] - ---- loginapp is running ----
 DEBUG root [2019-09-14 17:09:35,164] - Components::process(): Request for the process of identity...
  WARN root [2019-09-14 17:09:41,283] - Forwarding logs to logger[192.168.56.1:52067/0/0/0]...
 ERROR root [2019-09-14 17:10:46,667] - Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=46559325197, lastReceivedTime=46530470307, inactivityExceptionPeriod=134887564)
  INFO root [2019-09-14 17:10:46,670] - The following logs sent to logger failed:
  WARN root [2019-09-14 17:10:46,671] - ==> [2019-09-14 17:10:46 662] TCPPacketReceiver::processPendingEvents(192.168.56.1:52067): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-14 17:10:46,671] - ==> [2019-09-14 17:10:46 668] Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=46559325197, lastReceivedTime=46530470307, inactivityExceptionPeriod=134887564)
  WARN root [2019-09-14 17:10:46,781] - TCPPacketReceiver::processPendingEvents(192.168.56.1:52068): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-14 17:10:46,791] - Components::removeComponentByChannel: dbmgr : 4000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=46559603183, lastReceivedTime=46530615569, inactivityExceptionPeriod=134887564)
  INFO root [2019-09-14 17:10:46,804] - Shutdowner::onShutdownBegin: shutting down(period=0, tickPeriod=1)
S_INFO root [2019-09-14 17:10:46,814] - onLoginAppShutDown()
  INFO root [2019-09-14 17:10:46,814] - Shutdowner::onShutdown: shutting down
  INFO root [2019-09-14 17:10:46,814] - Shutdowner::onShutdownEnd: shutting down
  INFO root [2019-09-14 17:10:46,828] - Script::uninstall(): is successfully!
 DEBUG root [2019-09-14 17:10:46,828] - ThreadPool::destroy(): starting size 2.
  WARN root [2019-09-14 17:10:47,130] - ThreadPool::destroy(): waiting for thread(2)[0xc03a38,0xc034f0,], try=1
  INFO root [2019-09-14 17:11:50,668] - 
  INFO root [2019-09-14 17:11:50,677] - -----------------------------------------------------------------------------------------


  INFO root [2019-09-14 17:11:50,678] - Resmgr::initialize: KBE_ROOT=D:/kbengine/
  INFO root [2019-09-14 17:11:50,679] - Resmgr::initialize: KBE_RES_PATH=D:/kbengine/kbe/res/;D:/kbengine/BattleOfBalls_assets/;D:/kbengine/BattleOfBalls_assets/scripts/;D:/kbengine/BattleOfBalls_assets/res/
  INFO root [2019-09-14 17:11:50,679] - Resmgr::initialize: KBE_BIN_PATH=D:/kbengine/kbe/bin/server/
  INFO root [2019-09-14 17:11:50,728] - NetworkInterface::initialize(EXTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:20013 )
  INFO root [2019-09-14 17:11:50,728] - NetworkInterface::initialize(EXTERNAL-TCP): address 192.168.56.1:20013, SOMAXCONN=128.
  INFO root [2019-09-14 17:11:50,732] - NetworkInterface::initialize(INTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:52110 )
  INFO root [2019-09-14 17:11:50,732] - NetworkInterface::initialize(INTERNAL-TCP): address 192.168.56.1:52110, SOMAXCONN=128.
  INFO root [2019-09-14 17:11:50,732] - server-configs:
  INFO root [2019-09-14 17:11:50,732] - 	internalTcpAddr : 192.168.56.1:52110
  INFO root [2019-09-14 17:11:50,732] - 	externalTcpAddr : 192.168.56.1:20013
  INFO root [2019-09-14 17:11:50,733] - 	componentID : 9000
  WARN root [2019-09-14 17:11:50,952] - invalid UID(0) <= 0, please check UID for environment! automatically set to 2082.
  INFO root [2019-09-14 17:11:52,457] - ---- loginapp Version: 2.4.5. ScriptVersion: 0.1.0. Pythoncore: 3.7.3. Protocol: CE9FF53F4E85387EE4455C7191269324. Config: Debug X86. Built: 09:58:16 May  8 2019. AppID: 9000. UID: 2082. PID: 15648 ----
  INFO root [2019-09-14 17:11:52,457] - Server error_name: Intel(R) Core(TM) i5-6300HQ CPU @ 2.30GHz with 2097151kB RAM
  INFO root [2019-09-14 17:11:52,457] - ThreadPool::createThreadPool: creating  threadpool...
  INFO root [2019-09-14 17:11:52,458] - ThreadPool::createThreadPool: successfully(2), newThreadCount=1, normalMaxThreadCount=2, maxThreadCount=8
  INFO root [2019-09-14 17:11:52,701] - Script::install(): is successfully, Python=(3.7.3 (default, May  8 2019, 09:53:38) [MSC v.1916 32 bit (Intel)])!
  INFO root [2019-09-14 17:11:52,701] - Script::install(): KBExtra is successfully!
S_INFO root [2019-09-14 17:11:52,713] - onLoginAppReady: bootstrapGroupIndex=0, bootstrapGlobalIndex=0
  INFO root [2019-09-14 17:11:52,714] - TelnetServer server is running on port 31000
  INFO root [2019-09-14 17:11:53,156] - ---- loginapp is running ----
 DEBUG root [2019-09-14 17:11:53,157] - Components::process(): Request for the process of identity...
  INFO root [2019-09-14 17:12:00,334] - Components::findComponents: find logger(1)...
  INFO root [2019-09-14 17:12:00,441] - Components::findComponents: found logger, addr:192.168.56.1:52104
  INFO root [2019-09-14 17:12:00,443] - Components::addComponent: reset orderLog, uid=2082!
  INFO root [2019-09-14 17:12:00,444] - Components::addComponent[logger], uid=2082, componentID=2000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2019-09-14 17:12:00,448] - Components::findComponents: find dbmgr(2)...
  INFO root [2019-09-14 17:12:00,551] - Components::findComponents: found dbmgr, addr:192.168.56.1:52105
  INFO root [2019-09-14 17:12:00,551] - Components::addComponent: reset orderLog, uid=2082!
  INFO root [2019-09-14 17:12:00,551] - Components::addComponent[dbmgr], uid=2082, componentID=4000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2019-09-14 17:12:00,551] - Components::findComponents: found dbmgr, addr:192.168.56.1:52105
  WARN root [2019-09-14 17:12:01,552] - Forwarding logs to logger[192.168.56.1:52104/0/0/0]...
 ERROR root [2019-09-14 17:13:09,191] - Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=46880001691, lastReceivedTime=46838208695, inactivityExceptionPeriod=134887437)
  INFO root [2019-09-14 17:13:09,197] - The following logs sent to logger failed:
  WARN root [2019-09-14 17:13:09,197] - ==> [2019-09-14 17:13:09 190] TCPPacketReceiver::processPendingEvents(192.168.56.1:52104): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-14 17:13:09,197] - ==> [2019-09-14 17:13:09 192] Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=46880001691, lastReceivedTime=46838208695, inactivityExceptionPeriod=134887437)
  WARN root [2019-09-14 17:13:09,302] - TCPPacketReceiver::processPendingEvents(192.168.56.1:52105): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-14 17:13:09,302] - Components::removeComponentByChannel: dbmgr : 4000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=46880252944, lastReceivedTime=46838354489, inactivityExceptionPeriod=134887437)
  INFO root [2019-09-14 17:13:09,315] - Shutdowner::onShutdownBegin: shutting down(period=0, tickPeriod=1)
S_INFO root [2019-09-14 17:13:09,326] - onLoginAppShutDown()
  INFO root [2019-09-14 17:13:09,327] - Shutdowner::onShutdown: shutting down
  INFO root [2019-09-14 17:13:09,327] - Shutdowner::onShutdownEnd: shutting down
  INFO root [2019-09-14 17:13:09,347] - Script::uninstall(): is successfully!
 DEBUG root [2019-09-14 17:13:09,348] - ThreadPool::destroy(): starting size 2.
  WARN root [2019-09-14 17:13:09,650] - ThreadPool::destroy(): waiting for thread(2)[0x8303d0,0x8305d8,], try=1
